// angular-epic-spinners/angular-epic-spinners.es5.js
import { Component, Input, NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
var AtomSpinnerComponent = /** @class */ (function () {
    function AtomSpinnerComponent() {
        this.size = 60;
        this.animationDuration = 1000;
        this.color = 'red';
    }
    Object.defineProperty(AtomSpinnerComponent.prototype, "spinnerStyle", {
        /**
         * @return {?}
         */
        get: function () {
            return {
                height: this.size + "px",
                width: this.size + "px"
            };
        },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(AtomSpinnerComponent.prototype, "circleStyle", {
        /**
         * @return {?}
         */
        get: function () {
            return {
                color: this.color,
                fontSize: this.size * 0.24 + "px"
            };
        },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(AtomSpinnerComponent.prototype, "lineStyle", {
        /**
         * @return {?}
         */
        get: function () {
            return {
                animationDuration: this.animationDuration + "ms",
                borderLeftWidth: this.size / 25 + "px",
                borderRightWidth: this.size / 25 + "px",
                borderLeftColor: this.color
            };
        },
        enumerable: true,
        configurable: true
    });
    /**
     * @return {?}
     */
    AtomSpinnerComponent.prototype.ngOnInit = function () {
    };
    return AtomSpinnerComponent;
}());
AtomSpinnerComponent.decorators = [
    { type: Component, args: [{
                selector: 'app-atom-spinner',
                template: "\n      <div class=\"atom-spinner\" [ngStyle]=\"spinnerStyle\">\n        <div class=\"spinner-inner\">\n          <div class=\"spinner-line\" [ngStyle]=\"lineStyle\"></div>\n          <div class=\"spinner-line\" [ngStyle]=\"lineStyle\"></div>\n          <div class=\"spinner-line\" [ngStyle]=\"lineStyle\"></div>\n          <div class=\"spinner-circle\" [ngStyle]=\"circleStyle\">\n            &#9679;\n          </div>\n        </div>\n      </div>\n  ",
                styles: ["\n    .atom-spinner,\n    .atom-spinner * {\n      -webkit-box-sizing: border-box;\n              box-sizing: border-box;\n    }\n\n    .atom-spinner {\n      height: 60px;\n      width: 60px;\n      overflow: hidden;\n    }\n\n    .atom-spinner .spinner-inner {\n      position: relative;\n      display: block;\n      height: 100%;\n      width: 100%;\n    }\n\n    .atom-spinner .spinner-circle {\n      display: block;\n      position: absolute;\n      color: #ff1d5e;\n      font-size: calc(60px * 0.24);\n      top: 50%;\n      left: 50%;\n      -webkit-transform: translate(-50%, -50%);\n              transform: translate(-50%, -50%);\n    }\n\n    .atom-spinner .spinner-line {\n      position: absolute;\n      width: 100%;\n      height: 100%;\n      border-radius: 50%;\n      -webkit-animation-duration: 1s;\n              animation-duration: 1s;\n      border-left-width: calc(60px / 25);\n      border-top-width: calc(60px / 25);\n      border-left-color: #ff1d5e;\n      border-left-style: solid;\n      border-top-style: solid;\n      border-top-color: transparent;\n    }\n\n    .atom-spinner .spinner-line:nth-child(1) {\n      -webkit-animation: atom-spinner-animation-1 1s linear infinite;\n              animation: atom-spinner-animation-1 1s linear infinite;\n      -webkit-transform: rotateZ(120deg) rotateX(66deg) rotateZ(0deg);\n              transform: rotateZ(120deg) rotateX(66deg) rotateZ(0deg);\n    }\n\n    .atom-spinner .spinner-line:nth-child(2) {\n      -webkit-animation: atom-spinner-animation-2 1s linear infinite;\n              animation: atom-spinner-animation-2 1s linear infinite;\n      -webkit-transform: rotateZ(240deg) rotateX(66deg) rotateZ(0deg);\n              transform: rotateZ(240deg) rotateX(66deg) rotateZ(0deg);\n    }\n\n    .atom-spinner .spinner-line:nth-child(3) {\n      -webkit-animation: atom-spinner-animation-3 1s linear infinite;\n              animation: atom-spinner-animation-3 1s linear infinite;\n      -webkit-transform: rotateZ(360deg) rotateX(66deg) rotateZ(0deg);\n              transform: rotateZ(360deg) rotateX(66deg) rotateZ(0deg);\n    }\n\n    @-webkit-keyframes atom-spinner-animation-1 {\n      100% {\n        -webkit-transform: rotateZ(120deg) rotateX(66deg) rotateZ(360deg);\n                transform: rotateZ(120deg) rotateX(66deg) rotateZ(360deg);\n      }\n    }\n\n    @keyframes atom-spinner-animation-1 {\n      100% {\n        -webkit-transform: rotateZ(120deg) rotateX(66deg) rotateZ(360deg);\n                transform: rotateZ(120deg) rotateX(66deg) rotateZ(360deg);\n      }\n    }\n\n    @-webkit-keyframes atom-spinner-animation-2 {\n      100% {\n        -webkit-transform: rotateZ(240deg) rotateX(66deg) rotateZ(360deg);\n                transform: rotateZ(240deg) rotateX(66deg) rotateZ(360deg);\n      }\n    }\n\n    @keyframes atom-spinner-animation-2 {\n      100% {\n        -webkit-transform: rotateZ(240deg) rotateX(66deg) rotateZ(360deg);\n                transform: rotateZ(240deg) rotateX(66deg) rotateZ(360deg);\n      }\n    }\n\n    @-webkit-keyframes atom-spinner-animation-3 {\n      100% {\n        -webkit-transform: rotateZ(360deg) rotateX(66deg) rotateZ(360deg);\n                transform: rotateZ(360deg) rotateX(66deg) rotateZ(360deg);\n      }\n    }\n\n    @keyframes atom-spinner-animation-3 {\n      100% {\n        -webkit-transform: rotateZ(360deg) rotateX(66deg) rotateZ(360deg);\n                transform: rotateZ(360deg) rotateX(66deg) rotateZ(360deg);\n      }\n    }\n  "]
            },] },
];
/**
 * @nocollapse
 */
AtomSpinnerComponent.ctorParameters = function () { return []; };
AtomSpinnerComponent.propDecorators = {
    'size': [{ type: Input },],
    'animationDuration': [{ type: Input },],
    'color': [{ type: Input },],
};
var AtomSpinnerModule = /** @class */ (function () {
    function AtomSpinnerModule() {
    }
    return AtomSpinnerModule;
}());
AtomSpinnerModule.decorators = [
    { type: NgModule, args: [{
                imports: [CommonModule],
                exports: [AtomSpinnerComponent],
                declarations: [AtomSpinnerComponent],
                providers: [],
            },] },
];
/**
 * @nocollapse
 */
AtomSpinnerModule.ctorParameters = function () { return []; };
var BreedingRhombusSpinnerComponent = /** @class */ (function () {
    function BreedingRhombusSpinnerComponent() {
        this.size = 150;
        this.animationDuration = 2000;
        this.color = '#fff';
        this.animationBaseName = 'breeding-rhombus-spinner-animation-child';
        this.rhombusesNum = 8;
    }
    Object.defineProperty(BreedingRhombusSpinnerComponent.prototype, "spinnerStyle", {
        /**
         * @return {?}
         */
        get: function () {
            return {
                height: this.size + "px",
                width: this.size + "px"
            };
        },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(BreedingRhombusSpinnerComponent.prototype, "rhombusStyle", {
        /**
         * @return {?}
         */
        get: function () {
            return {
                height: this.size / 7.5 + "px",
                width: this.size / 7.5 + "px",
                animationDuration: "" + this.animationDuration,
                top: this.size / 2.3077 + "px",
                left: this.size / 23077 + "px",
                backgroundColor: this.color
            };
        },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(BreedingRhombusSpinnerComponent.prototype, "rhombusesStyles", {
        /**
         * @return {?}
         */
        get: function () {
            var /** @type {?} */ rhombusesStyles = [];
            var /** @type {?} */ delayModifier = this.animationDuration * 0.05;
            for (var /** @type {?} */ i = 1; i <= this.rhombusesNum; i++) {
                rhombusesStyles.push(Object.assign({}, this.rhombusStyle, { animationDelay: delayModifier * (i + 1) + "ms" }));
            }
            return rhombusesStyles;
        },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(BreedingRhombusSpinnerComponent.prototype, "bigRhombusStyle", {
        /**
         * @return {?}
         */
        get: function () {
            return {
                height: this.size / 3 + "px",
                width: this.size / 3 + "px",
                animationDuration: "" + this.animationDuration,
                top: this.size / 3 + "px",
                left: this.size / 3 + "px",
                backgroundColor: this.color
            };
        },
        enumerable: true,
        configurable: true
    });
    /**
     * @param {?} index
     * @return {?}
     */
    BreedingRhombusSpinnerComponent.prototype.childClassString = function (index) {
        return "child-" + (index + 1);
    };
    /**
     * @return {?}
     */
    BreedingRhombusSpinnerComponent.prototype.ngOnInit = function () {
    };
    return BreedingRhombusSpinnerComponent;
}());
BreedingRhombusSpinnerComponent.decorators = [
    { type: Component, args: [{
                selector: 'app-breeding-rhombus-spinner',
                template: "\n      <div class=\"breeding-rhombus-spinner\" [ngStyle]=\"spinnerStyle\">\n        <div class=\"rhombus\" *ngFor=\"let rs of rhombusesStyles; let i = index;\" [ngStyle]=\"rs\" [ngClass]=\"childClassString(i)\"></div>\n        <div class=\"rhombus big\" [ngStyle]=\"bigRhombusStyle\"></div>\n      </div>\n    ",
                styles: ["\n     .breeding-rhombus-spinner {\n       height: 65px;\n       width: 65px;\n       position: relative;\n       -webkit-transform: rotate(45deg);\n               transform: rotate(45deg);\n     }\n\n     .breeding-rhombus-spinner,\n     .breeding-rhombus-spinner * {\n       -webkit-box-sizing: border-box;\n               box-sizing: border-box;\n     }\n\n     .breeding-rhombus-spinner .rhombus {\n       height: calc(65px / 7.5);\n       width: calc(65px / 7.5);\n       -webkit-animation-duration: 2000ms;\n               animation-duration: 2000ms;\n       top: calc(65px / 2.3077);\n       left: calc(65px / 2.3077);\n       background-color: #ff1d5e;\n       position: absolute;\n       -webkit-animation-iteration-count: infinite;\n               animation-iteration-count: infinite;\n     }\n\n     .breeding-rhombus-spinner .rhombus:nth-child(2n+0) {\n       margin-right: 0;\n     }\n\n     .breeding-rhombus-spinner .rhombus.child-1 {\n       -webkit-animation-name: breeding-rhombus-spinner-animation-child-1;\n               animation-name: breeding-rhombus-spinner-animation-child-1;\n       -webkit-animation-delay: calc(100ms * 1);\n               animation-delay: calc(100ms * 1);\n     }\n\n     .breeding-rhombus-spinner .rhombus.child-2 {\n       -webkit-animation-name: breeding-rhombus-spinner-animation-child-2;\n               animation-name: breeding-rhombus-spinner-animation-child-2;\n       -webkit-animation-delay: calc(100ms * 2);\n               animation-delay: calc(100ms * 2);\n     }\n\n     .breeding-rhombus-spinner .rhombus.child-3 {\n       -webkit-animation-name: breeding-rhombus-spinner-animation-child-3;\n               animation-name: breeding-rhombus-spinner-animation-child-3;\n       -webkit-animation-delay: calc(100ms * 3);\n               animation-delay: calc(100ms * 3);\n     }\n\n     .breeding-rhombus-spinner .rhombus.child-4 {\n       -webkit-animation-name: breeding-rhombus-spinner-animation-child-4;\n               animation-name: breeding-rhombus-spinner-animation-child-4;\n       -webkit-animation-delay: calc(100ms * 4);\n               animation-delay: calc(100ms * 4);\n     }\n\n     .breeding-rhombus-spinner .rhombus.child-5 {\n       -webkit-animation-name: breeding-rhombus-spinner-animation-child-5;\n               animation-name: breeding-rhombus-spinner-animation-child-5;\n       -webkit-animation-delay: calc(100ms * 5);\n               animation-delay: calc(100ms * 5);\n     }\n\n     .breeding-rhombus-spinner .rhombus.child-6 {\n       -webkit-animation-name: breeding-rhombus-spinner-animation-child-6;\n               animation-name: breeding-rhombus-spinner-animation-child-6;\n       -webkit-animation-delay: calc(100ms * 6);\n               animation-delay: calc(100ms * 6);\n     }\n\n     .breeding-rhombus-spinner .rhombus.child-7 {\n       -webkit-animation-name: breeding-rhombus-spinner-animation-child-7;\n               animation-name: breeding-rhombus-spinner-animation-child-7;\n       -webkit-animation-delay: calc(100ms * 7);\n               animation-delay: calc(100ms * 7);\n     }\n\n     .breeding-rhombus-spinner .rhombus.child-8 {\n       -webkit-animation-name: breeding-rhombus-spinner-animation-child-8;\n               animation-name: breeding-rhombus-spinner-animation-child-8;\n       -webkit-animation-delay: calc(100ms * 8);\n               animation-delay: calc(100ms * 8);\n     }\n\n     .breeding-rhombus-spinner .rhombus.big {\n       height: calc(65px / 3);\n       width: calc(65px / 3);\n       -webkit-animation-duration: 2000ms;\n               animation-duration: 2000ms;\n       top: calc(65px / 3);\n       left: calc(65px / 3);\n       background-color: #ff1d5e;\n       -webkit-animation: breeding-rhombus-spinner-animation-child-big 2s infinite;\n               animation: breeding-rhombus-spinner-animation-child-big 2s infinite;\n       -webkit-animation-delay: 0.5s;\n               animation-delay: 0.5s;\n     }\n\n     @-webkit-keyframes breeding-rhombus-spinner-animation-child-1 {\n       50% {\n         -webkit-transform: translate(-325%, -325%);\n                 transform: translate(-325%, -325%);\n       }\n     }\n\n     @keyframes breeding-rhombus-spinner-animation-child-1 {\n       50% {\n         -webkit-transform: translate(-325%, -325%);\n                 transform: translate(-325%, -325%);\n       }\n     }\n\n     @-webkit-keyframes breeding-rhombus-spinner-animation-child-2 {\n       50% {\n         -webkit-transform: translate(0, -325%);\n                 transform: translate(0, -325%);\n       }\n     }\n\n     @keyframes breeding-rhombus-spinner-animation-child-2 {\n       50% {\n         -webkit-transform: translate(0, -325%);\n                 transform: translate(0, -325%);\n       }\n     }\n\n     @-webkit-keyframes breeding-rhombus-spinner-animation-child-3 {\n       50% {\n         -webkit-transform: translate(325%, -325%);\n                 transform: translate(325%, -325%);\n       }\n     }\n\n     @keyframes breeding-rhombus-spinner-animation-child-3 {\n       50% {\n         -webkit-transform: translate(325%, -325%);\n                 transform: translate(325%, -325%);\n       }\n     }\n\n     @-webkit-keyframes breeding-rhombus-spinner-animation-child-4 {\n       50% {\n         -webkit-transform: translate(325%, 0);\n                 transform: translate(325%, 0);\n       }\n     }\n\n     @keyframes breeding-rhombus-spinner-animation-child-4 {\n       50% {\n         -webkit-transform: translate(325%, 0);\n                 transform: translate(325%, 0);\n       }\n     }\n\n     @-webkit-keyframes breeding-rhombus-spinner-animation-child-5 {\n       50% {\n         -webkit-transform: translate(325%, 325%);\n                 transform: translate(325%, 325%);\n       }\n     }\n\n     @keyframes breeding-rhombus-spinner-animation-child-5 {\n       50% {\n         -webkit-transform: translate(325%, 325%);\n                 transform: translate(325%, 325%);\n       }\n     }\n\n     @-webkit-keyframes breeding-rhombus-spinner-animation-child-6 {\n       50% {\n         -webkit-transform: translate(0, 325%);\n                 transform: translate(0, 325%);\n       }\n     }\n\n     @keyframes breeding-rhombus-spinner-animation-child-6 {\n       50% {\n         -webkit-transform: translate(0, 325%);\n                 transform: translate(0, 325%);\n       }\n     }\n\n     @-webkit-keyframes breeding-rhombus-spinner-animation-child-7 {\n       50% {\n         -webkit-transform: translate(-325%, 325%);\n                 transform: translate(-325%, 325%);\n       }\n     }\n\n     @keyframes breeding-rhombus-spinner-animation-child-7 {\n       50% {\n         -webkit-transform: translate(-325%, 325%);\n                 transform: translate(-325%, 325%);\n       }\n     }\n\n     @-webkit-keyframes breeding-rhombus-spinner-animation-child-8 {\n       50% {\n         -webkit-transform: translate(-325%, 0);\n                 transform: translate(-325%, 0);\n       }\n     }\n\n     @keyframes breeding-rhombus-spinner-animation-child-8 {\n       50% {\n         -webkit-transform: translate(-325%, 0);\n                 transform: translate(-325%, 0);\n       }\n     }\n\n     @-webkit-keyframes breeding-rhombus-spinner-animation-child-big {\n       50% {\n         -webkit-transform: scale(0.5);\n                 transform: scale(0.5);\n       }\n     }\n\n     @keyframes breeding-rhombus-spinner-animation-child-big {\n       50% {\n         -webkit-transform: scale(0.5);\n                 transform: scale(0.5);\n       }\n     }\n  "]
            },] },
];
/**
 * @nocollapse
 */
BreedingRhombusSpinnerComponent.ctorParameters = function () { return []; };
BreedingRhombusSpinnerComponent.propDecorators = {
    'size': [{ type: Input },],
    'animationDuration': [{ type: Input },],
    'color': [{ type: Input },],
};
var BreedingRhombusSpinnerModule = /** @class */ (function () {
    function BreedingRhombusSpinnerModule() {
    }
    return BreedingRhombusSpinnerModule;
}());
BreedingRhombusSpinnerModule.decorators = [
    { type: NgModule, args: [{
                imports: [CommonModule],
                exports: [BreedingRhombusSpinnerComponent],
                declarations: [BreedingRhombusSpinnerComponent],
                providers: [],
            },] },
];
/**
 * @nocollapse
 */
BreedingRhombusSpinnerModule.ctorParameters = function () { return []; };
var CirclesToRhumbusesSpinnerComponent = /** @class */ (function () {
    function CirclesToRhumbusesSpinnerComponent() {
        this.circleSize = 15;
        this.circlesNum = 3;
        this.animationDuration = 1200;
        this.color = '#fff';
    }
    Object.defineProperty(CirclesToRhumbusesSpinnerComponent.prototype, "circleMarginLeft", {
        /**
         * @return {?}
         */
        get: function () {
            return this.circleSize * 1.125;
        },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(CirclesToRhumbusesSpinnerComponent.prototype, "spinnerStyle", {
        /**
         * @return {?}
         */
        get: function () {
            return {
                height: this.circleSize + "px",
                width: (this.circleSize + this.circleMarginLeft) * this.circlesNum + "px"
            };
        },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(CirclesToRhumbusesSpinnerComponent.prototype, "circleStyle", {
        /**
         * @return {?}
         */
        get: function () {
            return {
                borderColor: this.color,
                animationDuration: this.animationDuration + "ms",
                height: this.circleSize + "px",
                width: this.circleSize + "px",
                marginLeft: this.circleMarginLeft + "px"
            };
        },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(CirclesToRhumbusesSpinnerComponent.prototype, "circlesStyles", {
        /**
         * @return {?}
         */
        get: function () {
            var /** @type {?} */ circlesStyles = [];
            var /** @type {?} */ delay = 150;
            for (var /** @type {?} */ i = 1; i <= this.circlesNum; i++) {
                var /** @type {?} */ style = Object.assign({}, this.circleStyle, { animationDelay: i * delay + "ms" });
                circlesStyles.push(style);
            }
            return circlesStyles;
        },
        enumerable: true,
        configurable: true
    });
    /**
     * @return {?}
     */
    CirclesToRhumbusesSpinnerComponent.prototype.ngOnInit = function () {
    };
    return CirclesToRhumbusesSpinnerComponent;
}());
CirclesToRhumbusesSpinnerComponent.decorators = [
    { type: Component, args: [{
                selector: 'app-circles-to-rhumbuses-spinner',
                template: "\n    <div class=\"circles-to-rhombuses-spinner\" [ngStyle]=\"spinnerStyle\">\n      <div class=\"circle\" *ngFor=\"let cs of circlesStyles; let i = index;\" [ngStyle]=\"cs\"></div>\n    </div>\n  ",
                styles: ["\n    .circles-to-rhombuses-spinner, .circles-to-rhombuses-spinner * {\n      -webkit-box-sizing: border-box;\n              box-sizing: border-box;\n    }\n\n    .circles-to-rhombuses-spinner {\n      height: 15px;\n      width: calc( (15px + 15px * 1.125) * 3);\n      display: -webkit-box;\n      display: -ms-flexbox;\n      display: flex;\n      -webkit-box-align: center;\n          -ms-flex-align: center;\n              align-items: center;\n      -webkit-box-pack: center;\n          -ms-flex-pack: center;\n              justify-content: center\n    }\n\n    .circles-to-rhombuses-spinner .circle {\n      height: 15px;\n      width: 15px;\n      margin-left: calc(15px * 1.125);\n      -webkit-transform: rotate(45deg);\n              transform: rotate(45deg);\n      border-radius: 10%;\n      border: 3px solid #ff1d5e;\n      overflow: hidden;\n      background: transparent;\n\n      -webkit-animation: circles-to-rhombuses-animation 1200ms linear infinite;\n\n              animation: circles-to-rhombuses-animation 1200ms linear infinite;\n    }\n\n    .circles-to-rhombuses-spinner .circle:nth-child(1) {\n      -webkit-animation-delay: calc(150ms * 1);\n              animation-delay: calc(150ms * 1);\n      margin-left: 0\n    }\n\n    .circles-to-rhombuses-spinner .circle:nth-child(2) {\n      -webkit-animation-delay: calc(150ms * 2);\n              animation-delay: calc(150ms * 2);\n    }\n\n    .circles-to-rhombuses-spinner .circle:nth-child(3) {\n      -webkit-animation-delay: calc(150ms * 3);\n              animation-delay: calc(150ms * 3);\n    }\n\n    @-webkit-keyframes circles-to-rhombuses-animation {\n      0% {\n        border-radius: 10%;\n      }\n\n      17.5% {\n        border-radius: 10%;\n      }\n\n      50% {\n        border-radius: 100%;\n      }\n\n\n      93.5% {\n        border-radius: 10%;\n      }\n\n      100% {\n        border-radius: 10%;\n      }\n    }\n\n    @keyframes circles-to-rhombuses-animation {\n      0% {\n        border-radius: 10%;\n      }\n\n      17.5% {\n        border-radius: 10%;\n      }\n\n      50% {\n        border-radius: 100%;\n      }\n\n\n      93.5% {\n        border-radius: 10%;\n      }\n\n      100% {\n        border-radius: 10%;\n      }\n    }\n\n    @-webkit-keyframes circles-to-rhombuses-background-animation {\n      50% {\n        opacity: 0.4;\n      }\n    }\n\n    @keyframes circles-to-rhombuses-background-animation {\n      50% {\n        opacity: 0.4;\n      }\n    }\n  "]
            },] },
];
/**
 * @nocollapse
 */
CirclesToRhumbusesSpinnerComponent.ctorParameters = function () { return []; };
CirclesToRhumbusesSpinnerComponent.propDecorators = {
    'circleSize': [{ type: Input },],
    'circlesNum': [{ type: Input },],
    'animationDuration': [{ type: Input },],
    'color': [{ type: Input },],
};
var CirclesToRhumbusesSpinnerModule = /** @class */ (function () {
    function CirclesToRhumbusesSpinnerModule() {
    }
    return CirclesToRhumbusesSpinnerModule;
}());
CirclesToRhumbusesSpinnerModule.decorators = [
    { type: NgModule, args: [{
                imports: [CommonModule],
                exports: [CirclesToRhumbusesSpinnerComponent],
                declarations: [CirclesToRhumbusesSpinnerComponent],
                providers: [],
            },] },
];
/**
 * @nocollapse
 */
CirclesToRhumbusesSpinnerModule.ctorParameters = function () { return []; };
var FingerprintSpinnerComponent = /** @class */ (function () {
    function FingerprintSpinnerComponent() {
        this.size = 60;
        this.animationDuration = 1500;
        this.color = '#fff';
        this.ringsNum = 9;
        this.containerPadding = 2;
    }
    Object.defineProperty(FingerprintSpinnerComponent.prototype, "outerRingSize", {
        /**
         * @return {?}
         */
        get: function () {
            return this.size - this.containerPadding * 2;
        },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(FingerprintSpinnerComponent.prototype, "spinnerStyle", {
        /**
         * @return {?}
         */
        get: function () {
            return {
                height: this.size + "px",
                width: this.size + "px",
                padding: this.containerPadding + "px"
            };
        },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(FingerprintSpinnerComponent.prototype, "ringStyle", {
        /**
         * @return {?}
         */
        get: function () {
            return {
                borderTopColor: this.color,
                animationDuration: this.animationDuration + "ms"
            };
        },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(FingerprintSpinnerComponent.prototype, "ringsStyles", {
        /**
         * @return {?}
         */
        get: function () {
            var /** @type {?} */ ringsStyles = [];
            var /** @type {?} */ ringBase = this.outerRingSize / (this.ringsNum);
            var /** @type {?} */ ringInc = ringBase;
            for (var /** @type {?} */ i = 1; i <= this.ringsNum; i++) {
                ringsStyles.push(Object.assign({}, this.ringStyle, { animationDelay: i * 50 + "ms", height: ringBase + (i + 1) * ringInc + "px", width: ringBase + (i + 1) * ringInc + "px" }));
            }
            return ringsStyles;
        },
        enumerable: true,
        configurable: true
    });
    /**
     * @return {?}
     */
    FingerprintSpinnerComponent.prototype.ngOnInit = function () {
    };
    return FingerprintSpinnerComponent;
}());
FingerprintSpinnerComponent.decorators = [
    { type: Component, args: [{
                selector: 'app-fingerprint-spinner',
                template: "\n    <div class=\"fingerprint-spinner\" [ngStyle]=\"spinnerStyle\">\n      <div class=\"spinner-ring\" *ngFor=\"let rs of ringsStyles\" [ngStyle]=\"rs\"></div>\n    </div>\n\n  ",
                styles: ["\n    .fingerprint-spinner, .fingerprint-spinner * {\n      -webkit-box-sizing: border-box;\n              box-sizing: border-box;\n    }\n\n    .fingerprint-spinner {\n      height: 64px;\n      width: 64px;\n      padding: 2px;\n      overflow: hidden;\n      position: relative;\n    }\n\n    .fingerprint-spinner .spinner-ring {\n      position: absolute;\n      border-radius: 50%;\n      border: 2px solid transparent;\n      border-top-color: #ff1d5e;\n      -webkit-animation: fingerprint-spinner-animation 1500ms cubic-bezier(0.680, -0.750, 0.265, 1.750) infinite forwards;\n              animation: fingerprint-spinner-animation 1500ms cubic-bezier(0.680, -0.750, 0.265, 1.750) infinite forwards;\n      margin: auto;\n      bottom: 0;\n      left: 0;\n      right: 0;\n      top: 0;\n    }\n\n    .fingerprint-spinner .spinner-ring:nth-child(1) {\n      height: calc(60px / 9 + 0 * 60px / 9);\n      width: calc(60px / 9 + 0 * 60px / 9);\n      -webkit-animation-delay: calc(50ms * 1);\n              animation-delay: calc(50ms * 1);\n    }\n\n    .fingerprint-spinner .spinner-ring:nth-child(2) {\n      height: calc(60px / 9 + 1 * 60px / 9);\n      width: calc(60px / 9 + 1 * 60px / 9);\n      -webkit-animation-delay: calc(50ms * 2);\n              animation-delay: calc(50ms * 2);\n    }\n\n    .fingerprint-spinner .spinner-ring:nth-child(3) {\n      height: calc(60px / 9 + 2 * 60px / 9);\n      width: calc(60px / 9 + 2 * 60px / 9);\n      -webkit-animation-delay: calc(50ms * 3);\n              animation-delay: calc(50ms * 3);\n    }\n\n    .fingerprint-spinner .spinner-ring:nth-child(4) {\n      height: calc(60px / 9 + 3 * 60px / 9);\n      width: calc(60px / 9 + 3 * 60px / 9);\n      -webkit-animation-delay: calc(50ms * 4);\n              animation-delay: calc(50ms * 4);\n    }\n\n    .fingerprint-spinner .spinner-ring:nth-child(5) {\n      height: calc(60px / 9 + 4 * 60px / 9);\n      width: calc(60px / 9 + 4 * 60px / 9);\n      -webkit-animation-delay: calc(50ms * 5);\n              animation-delay: calc(50ms * 5);\n    }\n\n    .fingerprint-spinner .spinner-ring:nth-child(6) {\n      height: calc(60px / 9 + 5 * 60px / 9);\n      width: calc(60px / 9 + 5 * 60px / 9);\n      -webkit-animation-delay: calc(50ms * 6);\n              animation-delay: calc(50ms * 6);\n    }\n\n    .fingerprint-spinner .spinner-ring:nth-child(7) {\n      height: calc(60px / 9 + 6 * 60px / 9);\n      width: calc(60px / 9 + 6 * 60px / 9);\n      -webkit-animation-delay: calc(50ms * 7);\n              animation-delay: calc(50ms * 7);\n    }\n\n    .fingerprint-spinner .spinner-ring:nth-child(8) {\n      height: calc(60px / 9 + 7 * 60px / 9);\n      width: calc(60px / 9 + 7 * 60px / 9);\n      -webkit-animation-delay: calc(50ms * 8);\n              animation-delay: calc(50ms * 8);\n    }\n\n    .fingerprint-spinner .spinner-ring:nth-child(9) {\n      height: calc(60px / 9 + 8 * 60px / 9);\n      width: calc(60px / 9 + 8 * 60px / 9);\n      -webkit-animation-delay: calc(50ms * 9);\n              animation-delay: calc(50ms * 9);\n    }\n\n    @-webkit-keyframes fingerprint-spinner-animation {\n      100% {\n        -webkit-transform: rotate( 360deg );\n                transform: rotate( 360deg );\n      }\n    }\n\n    @keyframes fingerprint-spinner-animation {\n      100% {\n        -webkit-transform: rotate( 360deg );\n                transform: rotate( 360deg );\n      }\n    }\n  "]
            },] },
];
/**
 * @nocollapse
 */
FingerprintSpinnerComponent.ctorParameters = function () { return []; };
FingerprintSpinnerComponent.propDecorators = {
    'size': [{ type: Input },],
    'animationDuration': [{ type: Input },],
    'color': [{ type: Input },],
};
var FingerprintSpinnerModule = /** @class */ (function () {
    function FingerprintSpinnerModule() {
    }
    return FingerprintSpinnerModule;
}());
FingerprintSpinnerModule.decorators = [
    { type: NgModule, args: [{
                imports: [CommonModule],
                exports: [FingerprintSpinnerComponent],
                declarations: [FingerprintSpinnerComponent],
                providers: [],
            },] },
];
/**
 * @nocollapse
 */
FingerprintSpinnerModule.ctorParameters = function () { return []; };
var FulfillingBouncingCircleSpinnerComponent = /** @class */ (function () {
    function FulfillingBouncingCircleSpinnerComponent() {
        this.size = 60;
        this.animationDuration = 4000;
        this.color = '#fff';
    }
    Object.defineProperty(FulfillingBouncingCircleSpinnerComponent.prototype, "spinnerStyle", {
        /**
         * @return {?}
         */
        get: function () {
            return {
                height: this.size + "px",
                width: this.size + "px",
                animationDuration: this.animationDuration + "ms"
            };
        },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(FulfillingBouncingCircleSpinnerComponent.prototype, "orbitStyle", {
        /**
         * @return {?}
         */
        get: function () {
            return {
                height: this.size + "px",
                width: this.size + "px",
                borderColor: this.color,
                borderWidth: this.size * 0.03 + "px",
                animationDuration: this.animationDuration + "ms"
            };
        },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(FulfillingBouncingCircleSpinnerComponent.prototype, "circleStyle", {
        /**
         * @return {?}
         */
        get: function () {
            return {
                height: this.size + "px",
                width: this.size + "px",
                borderColor: this.color,
                color: this.color,
                borderWidth: this.size * 0.1 + "px",
                animationDuration: this.animationDuration + "ms"
            };
        },
        enumerable: true,
        configurable: true
    });
    /**
     * @return {?}
     */
    FulfillingBouncingCircleSpinnerComponent.prototype.ngOnInit = function () {
    };
    return FulfillingBouncingCircleSpinnerComponent;
}());
FulfillingBouncingCircleSpinnerComponent.decorators = [
    { type: Component, args: [{
                selector: 'app-fulfilling-bouncing-circle-spinner',
                template: "\n    <div class=\"fulfilling-bouncing-circle-spinner\" [ngStyle]=\"spinnerStyle\">\n      <div class=\"circle\" [ngStyle]=\"circleStyle\"></div>\n      <div class=\"orbit\" [ngStyle]=\"orbitStyle\"></div>\n    </div>\n  ",
                styles: ["\n\n    .fulfilling-bouncing-circle-spinner, .fulfilling-bouncing-circle-spinner * {\n      -webkit-box-sizing: border-box;\n              box-sizing: border-box;\n    }\n    .fulfilling-bouncing-circle-spinner {\n      height: 60px;\n      width: 60px;\n      position: relative;\n      -webkit-animation: fulfilling-bouncing-circle-spinner-animation infinite 4000ms ease;\n              animation: fulfilling-bouncing-circle-spinner-animation infinite 4000ms ease;\n    }\n    .fulfilling-bouncing-circle-spinner .orbit {\n      height: 60px;\n      width: 60px;\n      position: absolute;\n      top: 0;\n      left: 0;\n      border-radius: 50%;\n      border: calc(60px * 0.03) solid #ff1d5e;\n      -webkit-animation: fulfilling-bouncing-circle-spinner-orbit-animation infinite 4000ms ease;\n              animation: fulfilling-bouncing-circle-spinner-orbit-animation infinite 4000ms ease;\n    }\n    .fulfilling-bouncing-circle-spinner .circle {\n      height: 60px;\n      width: 60px;\n      color: #ff1d5e;\n      display: block;\n      border-radius: 50%;\n      position: relative;\n      border: calc(60px * 0.1) solid #ff1d5e;\n      -webkit-animation: fulfilling-bouncing-circle-spinner-circle-animation infinite 4000ms ease;\n              animation: fulfilling-bouncing-circle-spinner-circle-animation infinite 4000ms ease;\n      -webkit-transform: rotate(0deg) scale(1);\n              transform: rotate(0deg) scale(1);\n    }\n    @-webkit-keyframes fulfilling-bouncing-circle-spinner-animation {\n      0% {\n        -webkit-transform: rotate(0deg);\n                transform: rotate(0deg);\n      }\n      100% {\n        -webkit-transform: rotate(360deg);\n                transform: rotate(360deg);\n      }\n    }\n    @keyframes fulfilling-bouncing-circle-spinner-animation {\n      0% {\n        -webkit-transform: rotate(0deg);\n                transform: rotate(0deg);\n      }\n      100% {\n        -webkit-transform: rotate(360deg);\n                transform: rotate(360deg);\n      }\n    }\n    @-webkit-keyframes fulfilling-bouncing-circle-spinner-orbit-animation {\n      0% {\n        -webkit-transform: scale(1);\n                transform: scale(1);\n      }\n      50% {\n        -webkit-transform: scale(1);\n                transform: scale(1);\n      }\n      62.5% {\n        -webkit-transform: scale(0.8);\n                transform: scale(0.8);\n      }\n      75% {\n        -webkit-transform: scale(1);\n                transform: scale(1);\n      }\n      87.5% {\n        -webkit-transform: scale(0.8);\n                transform: scale(0.8);\n      }\n      100% {\n        -webkit-transform: scale(1);\n                transform: scale(1);\n      }\n    }\n    @keyframes fulfilling-bouncing-circle-spinner-orbit-animation {\n      0% {\n        -webkit-transform: scale(1);\n                transform: scale(1);\n      }\n      50% {\n        -webkit-transform: scale(1);\n                transform: scale(1);\n      }\n      62.5% {\n        -webkit-transform: scale(0.8);\n                transform: scale(0.8);\n      }\n      75% {\n        -webkit-transform: scale(1);\n                transform: scale(1);\n      }\n      87.5% {\n        -webkit-transform: scale(0.8);\n                transform: scale(0.8);\n      }\n      100% {\n        -webkit-transform: scale(1);\n                transform: scale(1);\n      }\n    }\n    @-webkit-keyframes fulfilling-bouncing-circle-spinner-circle-animation {\n      0% {\n        -webkit-transform: scale(1);\n                transform: scale(1);\n        border-color: transparent;\n        border-top-color: inherit;\n      }\n      16.7% {\n        border-color: transparent;\n        border-top-color: initial;\n        border-right-color: initial;\n      }\n      33.4% {\n        border-color: transparent;\n        border-top-color: inherit;\n        border-right-color: inherit;\n        border-bottom-color: inherit;\n      }\n      50% {\n        border-color: inherit;\n        -webkit-transform: scale(1);\n                transform: scale(1);\n      }\n      62.5% {\n        border-color: inherit;\n        -webkit-transform: scale(1.4);\n                transform: scale(1.4);\n      }\n      75% {\n        border-color: inherit;\n        -webkit-transform: scale(1);\n                transform: scale(1);\n        opacity: 1;\n      }\n      87.5% {\n        border-color: inherit;\n        -webkit-transform: scale(1.4);\n                transform: scale(1.4);\n      }\n      100% {\n        border-color: transparent;\n        border-top-color: inherit;\n        -webkit-transform: scale(1);\n                transform: scale(1);\n      }\n    }\n    @keyframes fulfilling-bouncing-circle-spinner-circle-animation {\n      0% {\n        -webkit-transform: scale(1);\n                transform: scale(1);\n        border-color: transparent;\n        border-top-color: inherit;\n      }\n      16.7% {\n        border-color: transparent;\n        border-top-color: initial;\n        border-right-color: initial;\n      }\n      33.4% {\n        border-color: transparent;\n        border-top-color: inherit;\n        border-right-color: inherit;\n        border-bottom-color: inherit;\n      }\n      50% {\n        border-color: inherit;\n        -webkit-transform: scale(1);\n                transform: scale(1);\n      }\n      62.5% {\n        border-color: inherit;\n        -webkit-transform: scale(1.4);\n                transform: scale(1.4);\n      }\n      75% {\n        border-color: inherit;\n        -webkit-transform: scale(1);\n                transform: scale(1);\n        opacity: 1;\n      }\n      87.5% {\n        border-color: inherit;\n        -webkit-transform: scale(1.4);\n                transform: scale(1.4);\n      }\n      100% {\n        border-color: transparent;\n        border-top-color: inherit;\n        -webkit-transform: scale(1);\n                transform: scale(1);\n      }\n    }\n  "]
            },] },
];
/**
 * @nocollapse
 */
FulfillingBouncingCircleSpinnerComponent.ctorParameters = function () { return []; };
FulfillingBouncingCircleSpinnerComponent.propDecorators = {
    'size': [{ type: Input },],
    'animationDuration': [{ type: Input },],
    'color': [{ type: Input },],
};
var FulfillingBouncingCircleSpinnerModule = /** @class */ (function () {
    function FulfillingBouncingCircleSpinnerModule() {
    }
    return FulfillingBouncingCircleSpinnerModule;
}());
FulfillingBouncingCircleSpinnerModule.decorators = [
    { type: NgModule, args: [{
                imports: [CommonModule],
                exports: [FulfillingBouncingCircleSpinnerComponent],
                declarations: [FulfillingBouncingCircleSpinnerComponent],
                providers: [],
            },] },
];
/**
 * @nocollapse
 */
FulfillingBouncingCircleSpinnerModule.ctorParameters = function () { return []; };
var FlowerSpinnerComponent = /** @class */ (function () {
    function FlowerSpinnerComponent() {
        this.size = 70;
        this.animationDuration = 2500;
        this.color = '#fff';
        this.smallerDotAnimationBaseName = 'flower-spinner-smaller-dot-animation';
        this.biggerDotAnimationBaseName = 'flower-spinner-bigger-dot-animation';
        this.currentSmallerDotAnimationBaseName = '';
        this.currentBiggerDotAnimationBaseName = '';
    }
    Object.defineProperty(FlowerSpinnerComponent.prototype, "dotSize", {
        /**
         * @return {?}
         */
        get: function () {
            return this.size / 7;
        },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(FlowerSpinnerComponent.prototype, "spinnerStyle", {
        /**
         * @return {?}
         */
        get: function () {
            return {
                height: this.size + "px",
                width: this.size + "px",
            };
        },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(FlowerSpinnerComponent.prototype, "dotsContainerStyle", {
        /**
         * @return {?}
         */
        get: function () {
            return {
                width: this.color + "px",
                height: this.color + "px",
            };
        },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(FlowerSpinnerComponent.prototype, "smallerDotStyle", {
        /**
         * @return {?}
         */
        get: function () {
            return {
                background: this.color,
                animationDuration: this.animationDuration + "ms",
                animationName: this.currentSmallerDotAnimationBaseName
            };
        },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(FlowerSpinnerComponent.prototype, "biggerDotStyle", {
        /**
         * @return {?}
         */
        get: function () {
            return {
                background: this.color,
                animationDuration: this.animationDuration + "ms",
                animationName: this.currentBiggerDotAnimationBaseName
            };
        },
        enumerable: true,
        configurable: true
    });
    /**
     * @return {?}
     */
    FlowerSpinnerComponent.prototype.ngOnInit = function () {
    };
    /**
     * @return {?}
     */
    FlowerSpinnerComponent.prototype.ngOnChanges = function () {
    };
    return FlowerSpinnerComponent;
}());
FlowerSpinnerComponent.decorators = [
    { type: Component, args: [{
                selector: 'app-flower-spinner',
                template: "\n    <div class=\"flower-spinner\" [ngStyle]=\"spinnerStyle\">\n      <div class=\"dots-container\" [ngStyle]=\"dotsContainerStyle\">\n        <div class=\"bigger-dot\" [ngStyle]=\"biggerDotStyle\">\n          <div class=\"smaller-dot\" [ngStyle]=\"smallerDotStyle\"></div>\n        </div>\n      </div>\n    </div>\n  ",
                styles: ["\n    .flower-spinner, .flower-spinner * {\n      -webkit-box-sizing: border-box;\n              box-sizing: border-box;\n    }\n\n    .flower-spinner {\n      height: 70px;\n      width: 70px;\n      display: -webkit-box;\n      display: -ms-flexbox;\n      display: flex;\n      -webkit-box-orient: horizontal;\n      -webkit-box-direction: normal;\n          -ms-flex-direction: row;\n              flex-direction: row;\n      -webkit-box-align: center;\n          -ms-flex-align: center;\n              align-items: center;\n      -webkit-box-pack: center;\n          -ms-flex-pack: center;\n              justify-content: center;\n    }\n\n    .flower-spinner .dots-container {\n      height: calc(70px / 7);\n      width: calc(70px / 7);\n    }\n\n    .flower-spinner .smaller-dot {\n      background: #ff1d5e;\n      height: 100%;\n      width: 100%;\n      border-radius: 50%;\n      -webkit-animation: flower-spinner-smaller-dot-animation 2.5s 0s infinite both;\n              animation: flower-spinner-smaller-dot-animation 2.5s 0s infinite both;\n    }\n\n    .flower-spinner .bigger-dot {\n      background: #ff1d5e;\n      height: 100%;\n      width: 100%;\n      padding: 10%;\n      border-radius: 50%;\n      -webkit-animation: flower-spinner-bigger-dot-animation 2.5s 0s infinite both;\n              animation: flower-spinner-bigger-dot-animation 2.5s 0s infinite both;\n    }\n\n    @-webkit-keyframes flower-spinner-bigger-dot-animation {\n      0%, 100% {\n        -webkit-box-shadow: rgb(255, 29, 94) 0 0 0,\n        rgb(255, 29, 94) 0 0 0,\n        rgb(255, 29, 94) 0 0 0,\n        rgb(255, 29, 94) 0 0 0,\n        rgb(255, 29, 94) 0 0 0,\n        rgb(255, 29, 94) 0 0 0,\n        rgb(255, 29, 94) 0 0 0,\n        rgb(255, 29, 94) 0 0 0;\n                box-shadow: rgb(255, 29, 94) 0 0 0,\n        rgb(255, 29, 94) 0 0 0,\n        rgb(255, 29, 94) 0 0 0,\n        rgb(255, 29, 94) 0 0 0,\n        rgb(255, 29, 94) 0 0 0,\n        rgb(255, 29, 94) 0 0 0,\n        rgb(255, 29, 94) 0 0 0,\n        rgb(255, 29, 94) 0 0 0;\n      }\n      50% {\n        -webkit-transform: rotate(180deg);\n                transform: rotate(180deg);\n      }\n      25%, 75% {\n        -webkit-box-shadow: rgb(255, 29, 94) 26px 0 0,\n        rgb(255, 29, 94) -26px 0 0,\n        rgb(255, 29, 94) 0 26px 0,\n        rgb(255, 29, 94) 0 -26px 0,\n        rgb(255, 29, 94) 19px -19px 0,\n        rgb(255, 29, 94) 19px 19px 0,\n        rgb(255, 29, 94) -19px -19px 0,\n        rgb(255, 29, 94) -19px 19px 0;\n                box-shadow: rgb(255, 29, 94) 26px 0 0,\n        rgb(255, 29, 94) -26px 0 0,\n        rgb(255, 29, 94) 0 26px 0,\n        rgb(255, 29, 94) 0 -26px 0,\n        rgb(255, 29, 94) 19px -19px 0,\n        rgb(255, 29, 94) 19px 19px 0,\n        rgb(255, 29, 94) -19px -19px 0,\n        rgb(255, 29, 94) -19px 19px 0;\n      }\n      100% {\n        -webkit-transform: rotate(360deg);\n                transform: rotate(360deg);\n        -webkit-box-shadow: rgb(255, 29, 94) 0 0 0,\n        rgb(255, 29, 94) 0 0 0,\n        rgb(255, 29, 94) 0 0 0,\n        rgb(255, 29, 94) 0 0 0,\n        rgb(255, 29, 94) 0 0 0,\n        rgb(255, 29, 94) 0 0 0,\n        rgb(255, 29, 94) 0 0 0,\n        rgb(255, 29, 94) 0 0 0;\n                box-shadow: rgb(255, 29, 94) 0 0 0,\n        rgb(255, 29, 94) 0 0 0,\n        rgb(255, 29, 94) 0 0 0,\n        rgb(255, 29, 94) 0 0 0,\n        rgb(255, 29, 94) 0 0 0,\n        rgb(255, 29, 94) 0 0 0,\n        rgb(255, 29, 94) 0 0 0,\n        rgb(255, 29, 94) 0 0 0;\n      }\n    }\n\n    @keyframes flower-spinner-bigger-dot-animation {\n      0%, 100% {\n        -webkit-box-shadow: rgb(255, 29, 94) 0 0 0,\n        rgb(255, 29, 94) 0 0 0,\n        rgb(255, 29, 94) 0 0 0,\n        rgb(255, 29, 94) 0 0 0,\n        rgb(255, 29, 94) 0 0 0,\n        rgb(255, 29, 94) 0 0 0,\n        rgb(255, 29, 94) 0 0 0,\n        rgb(255, 29, 94) 0 0 0;\n                box-shadow: rgb(255, 29, 94) 0 0 0,\n        rgb(255, 29, 94) 0 0 0,\n        rgb(255, 29, 94) 0 0 0,\n        rgb(255, 29, 94) 0 0 0,\n        rgb(255, 29, 94) 0 0 0,\n        rgb(255, 29, 94) 0 0 0,\n        rgb(255, 29, 94) 0 0 0,\n        rgb(255, 29, 94) 0 0 0;\n      }\n      50% {\n        -webkit-transform: rotate(180deg);\n                transform: rotate(180deg);\n      }\n      25%, 75% {\n        -webkit-box-shadow: rgb(255, 29, 94) 26px 0 0,\n        rgb(255, 29, 94) -26px 0 0,\n        rgb(255, 29, 94) 0 26px 0,\n        rgb(255, 29, 94) 0 -26px 0,\n        rgb(255, 29, 94) 19px -19px 0,\n        rgb(255, 29, 94) 19px 19px 0,\n        rgb(255, 29, 94) -19px -19px 0,\n        rgb(255, 29, 94) -19px 19px 0;\n                box-shadow: rgb(255, 29, 94) 26px 0 0,\n        rgb(255, 29, 94) -26px 0 0,\n        rgb(255, 29, 94) 0 26px 0,\n        rgb(255, 29, 94) 0 -26px 0,\n        rgb(255, 29, 94) 19px -19px 0,\n        rgb(255, 29, 94) 19px 19px 0,\n        rgb(255, 29, 94) -19px -19px 0,\n        rgb(255, 29, 94) -19px 19px 0;\n      }\n      100% {\n        -webkit-transform: rotate(360deg);\n                transform: rotate(360deg);\n        -webkit-box-shadow: rgb(255, 29, 94) 0 0 0,\n        rgb(255, 29, 94) 0 0 0,\n        rgb(255, 29, 94) 0 0 0,\n        rgb(255, 29, 94) 0 0 0,\n        rgb(255, 29, 94) 0 0 0,\n        rgb(255, 29, 94) 0 0 0,\n        rgb(255, 29, 94) 0 0 0,\n        rgb(255, 29, 94) 0 0 0;\n                box-shadow: rgb(255, 29, 94) 0 0 0,\n        rgb(255, 29, 94) 0 0 0,\n        rgb(255, 29, 94) 0 0 0,\n        rgb(255, 29, 94) 0 0 0,\n        rgb(255, 29, 94) 0 0 0,\n        rgb(255, 29, 94) 0 0 0,\n        rgb(255, 29, 94) 0 0 0,\n        rgb(255, 29, 94) 0 0 0;\n      }\n    }\n\n    @-webkit-keyframes flower-spinner-smaller-dot-animation {\n      0%, 100% {\n        -webkit-box-shadow: rgb(255, 29, 94) 0 0 0,\n        rgb(255, 29, 94) 0 0 0,\n        rgb(255, 29, 94) 0 0 0,\n        rgb(255, 29, 94) 0 0 0,\n        rgb(255, 29, 94) 0 0 0,\n        rgb(255, 29, 94) 0 0 0,\n        rgb(255, 29, 94) 0 0 0,\n        rgb(255, 29, 94) 0 0 0;\n                box-shadow: rgb(255, 29, 94) 0 0 0,\n        rgb(255, 29, 94) 0 0 0,\n        rgb(255, 29, 94) 0 0 0,\n        rgb(255, 29, 94) 0 0 0,\n        rgb(255, 29, 94) 0 0 0,\n        rgb(255, 29, 94) 0 0 0,\n        rgb(255, 29, 94) 0 0 0,\n        rgb(255, 29, 94) 0 0 0;\n      }\n      25%, 75% {\n        -webkit-box-shadow: rgb(255, 29, 94) 14px 0 0,\n        rgb(255, 29, 94) -14px 0 0,\n        rgb(255, 29, 94) 0 14px 0,\n        rgb(255, 29, 94) 0 -14px 0,\n        rgb(255, 29, 94) 10px -10px 0,\n        rgb(255, 29, 94) 10px 10px 0,\n        rgb(255, 29, 94) -10px -10px 0,\n        rgb(255, 29, 94) -10px 10px 0;\n                box-shadow: rgb(255, 29, 94) 14px 0 0,\n        rgb(255, 29, 94) -14px 0 0,\n        rgb(255, 29, 94) 0 14px 0,\n        rgb(255, 29, 94) 0 -14px 0,\n        rgb(255, 29, 94) 10px -10px 0,\n        rgb(255, 29, 94) 10px 10px 0,\n        rgb(255, 29, 94) -10px -10px 0,\n        rgb(255, 29, 94) -10px 10px 0;\n      }\n      100% {\n        -webkit-box-shadow: rgb(255, 29, 94) 0 0 0,\n        rgb(255, 29, 94) 0 0 0,\n        rgb(255, 29, 94) 0 0 0,\n        rgb(255, 29, 94) 0 0 0,\n        rgb(255, 29, 94) 0 0 0,\n        rgb(255, 29, 94) 0 0 0,\n        rgb(255, 29, 94) 0 0 0,\n        rgb(255, 29, 94) 0 0 0;\n                box-shadow: rgb(255, 29, 94) 0 0 0,\n        rgb(255, 29, 94) 0 0 0,\n        rgb(255, 29, 94) 0 0 0,\n        rgb(255, 29, 94) 0 0 0,\n        rgb(255, 29, 94) 0 0 0,\n        rgb(255, 29, 94) 0 0 0,\n        rgb(255, 29, 94) 0 0 0,\n        rgb(255, 29, 94) 0 0 0;\n      }\n    }\n\n    @keyframes flower-spinner-smaller-dot-animation {\n      0%, 100% {\n        -webkit-box-shadow: rgb(255, 29, 94) 0 0 0,\n        rgb(255, 29, 94) 0 0 0,\n        rgb(255, 29, 94) 0 0 0,\n        rgb(255, 29, 94) 0 0 0,\n        rgb(255, 29, 94) 0 0 0,\n        rgb(255, 29, 94) 0 0 0,\n        rgb(255, 29, 94) 0 0 0,\n        rgb(255, 29, 94) 0 0 0;\n                box-shadow: rgb(255, 29, 94) 0 0 0,\n        rgb(255, 29, 94) 0 0 0,\n        rgb(255, 29, 94) 0 0 0,\n        rgb(255, 29, 94) 0 0 0,\n        rgb(255, 29, 94) 0 0 0,\n        rgb(255, 29, 94) 0 0 0,\n        rgb(255, 29, 94) 0 0 0,\n        rgb(255, 29, 94) 0 0 0;\n      }\n      25%, 75% {\n        -webkit-box-shadow: rgb(255, 29, 94) 14px 0 0,\n        rgb(255, 29, 94) -14px 0 0,\n        rgb(255, 29, 94) 0 14px 0,\n        rgb(255, 29, 94) 0 -14px 0,\n        rgb(255, 29, 94) 10px -10px 0,\n        rgb(255, 29, 94) 10px 10px 0,\n        rgb(255, 29, 94) -10px -10px 0,\n        rgb(255, 29, 94) -10px 10px 0;\n                box-shadow: rgb(255, 29, 94) 14px 0 0,\n        rgb(255, 29, 94) -14px 0 0,\n        rgb(255, 29, 94) 0 14px 0,\n        rgb(255, 29, 94) 0 -14px 0,\n        rgb(255, 29, 94) 10px -10px 0,\n        rgb(255, 29, 94) 10px 10px 0,\n        rgb(255, 29, 94) -10px -10px 0,\n        rgb(255, 29, 94) -10px 10px 0;\n      }\n      100% {\n        -webkit-box-shadow: rgb(255, 29, 94) 0 0 0,\n        rgb(255, 29, 94) 0 0 0,\n        rgb(255, 29, 94) 0 0 0,\n        rgb(255, 29, 94) 0 0 0,\n        rgb(255, 29, 94) 0 0 0,\n        rgb(255, 29, 94) 0 0 0,\n        rgb(255, 29, 94) 0 0 0,\n        rgb(255, 29, 94) 0 0 0;\n                box-shadow: rgb(255, 29, 94) 0 0 0,\n        rgb(255, 29, 94) 0 0 0,\n        rgb(255, 29, 94) 0 0 0,\n        rgb(255, 29, 94) 0 0 0,\n        rgb(255, 29, 94) 0 0 0,\n        rgb(255, 29, 94) 0 0 0,\n        rgb(255, 29, 94) 0 0 0,\n        rgb(255, 29, 94) 0 0 0;\n      }\n    }\n  "]
            },] },
];
/**
 * @nocollapse
 */
FlowerSpinnerComponent.ctorParameters = function () { return []; };
FlowerSpinnerComponent.propDecorators = {
    'size': [{ type: Input },],
    'animationDuration': [{ type: Input },],
    'color': [{ type: Input },],
};
var FlowerSpinnerModule = /** @class */ (function () {
    function FlowerSpinnerModule() {
    }
    return FlowerSpinnerModule;
}());
FlowerSpinnerModule.decorators = [
    { type: NgModule, args: [{
                imports: [CommonModule],
                exports: [FlowerSpinnerComponent],
                declarations: [FlowerSpinnerComponent],
                providers: [],
            },] },
];
/**
 * @nocollapse
 */
FlowerSpinnerModule.ctorParameters = function () { return []; };
var FulfillingSquareSpinnerComponent = /** @class */ (function () {
    function FulfillingSquareSpinnerComponent() {
        this.animationDuration = 4000;
        this.size = 50;
        this.color = '#fff';
    }
    Object.defineProperty(FulfillingSquareSpinnerComponent.prototype, "spinnerStyle", {
        /**
         * @return {?}
         */
        get: function () {
            return {
                height: this.size + "px",
                width: this.size + "px",
                borderColor: this.color,
            };
        },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(FulfillingSquareSpinnerComponent.prototype, "spinnerInnerStyle", {
        /**
         * @return {?}
         */
        get: function () {
            return {
                backgroundColor: this.color
            };
        },
        enumerable: true,
        configurable: true
    });
    /**
     * @return {?}
     */
    FulfillingSquareSpinnerComponent.prototype.ngOnInit = function () {
    };
    return FulfillingSquareSpinnerComponent;
}());
FulfillingSquareSpinnerComponent.decorators = [
    { type: Component, args: [{
                selector: 'app-fulfilling-square-spinner',
                template: "\n    <div class=\"fulfilling-square-spinner\" [ngStyle]=\"spinnerStyle\">\n      <div class=\"spinner-inner\" [ngStyle]=\"spinnerInnerStyle\"></div>\n    </div>\n  ",
                styles: ["\n    .fulfilling-square-spinner, .fulfilling-square-spinner * {\n      -webkit-box-sizing: border-box;\n              box-sizing: border-box;\n    }\n    .fulfilling-square-spinner {\n      height: 50px;\n      width: 50px;\n      position: relative;\n      border: 4px solid #ff1d5e;\n      -webkit-animation: fulfilling-square-spinner-animation 4s infinite ease;\n              animation: fulfilling-square-spinner-animation 4s infinite ease;\n    }\n    .fulfilling-square-spinner .spinner-inner {\n      vertical-align: top;\n      display: inline-block;\n      background-color: #ff1d5e;\n      width: 100%;\n      opacity: 1;\n      -webkit-animation: fulfilling-square-spinner-inner-animation 4s infinite ease-in;\n              animation: fulfilling-square-spinner-inner-animation 4s infinite ease-in;\n    }\n    @-webkit-keyframes fulfilling-square-spinner-animation {\n      0% {\n        -webkit-transform: rotate(0deg);\n                transform: rotate(0deg);\n      }\n      25% {\n        -webkit-transform: rotate(180deg);\n                transform: rotate(180deg);\n      }\n      50% {\n        -webkit-transform: rotate(180deg);\n                transform: rotate(180deg);\n      }\n      75% {\n        -webkit-transform: rotate(360deg);\n                transform: rotate(360deg);\n      }\n      100% {\n        -webkit-transform: rotate(360deg);\n                transform: rotate(360deg);\n      }\n    }\n    @keyframes fulfilling-square-spinner-animation {\n      0% {\n        -webkit-transform: rotate(0deg);\n                transform: rotate(0deg);\n      }\n      25% {\n        -webkit-transform: rotate(180deg);\n                transform: rotate(180deg);\n      }\n      50% {\n        -webkit-transform: rotate(180deg);\n                transform: rotate(180deg);\n      }\n      75% {\n        -webkit-transform: rotate(360deg);\n                transform: rotate(360deg);\n      }\n      100% {\n        -webkit-transform: rotate(360deg);\n                transform: rotate(360deg);\n      }\n    }\n    @-webkit-keyframes fulfilling-square-spinner-inner-animation {\n      0% {\n        height: 0;\n      }\n      25% {\n        height: 0;\n      }\n      50% {\n        height: 100%;\n      }\n      75% {\n        height: 100%;\n      }\n      100% {\n        height: 0;\n      }\n    }\n    @keyframes fulfilling-square-spinner-inner-animation {\n      0% {\n        height: 0;\n      }\n      25% {\n        height: 0;\n      }\n      50% {\n        height: 100%;\n      }\n      75% {\n        height: 100%;\n      }\n      100% {\n        height: 0;\n      }\n    }\n  "]
            },] },
];
/**
 * @nocollapse
 */
FulfillingSquareSpinnerComponent.ctorParameters = function () { return []; };
FulfillingSquareSpinnerComponent.propDecorators = {
    'animationDuration': [{ type: Input },],
    'size': [{ type: Input },],
    'color': [{ type: Input },],
};
var FulfillingSquareSpinnerModule = /** @class */ (function () {
    function FulfillingSquareSpinnerModule() {
    }
    return FulfillingSquareSpinnerModule;
}());
FulfillingSquareSpinnerModule.decorators = [
    { type: NgModule, args: [{
                imports: [CommonModule],
                exports: [FulfillingSquareSpinnerComponent],
                declarations: [FulfillingSquareSpinnerComponent],
                providers: [],
            },] },
];
/**
 * @nocollapse
 */
FulfillingSquareSpinnerModule.ctorParameters = function () { return []; };
var HalfCircleSpinnerComponent = /** @class */ (function () {
    function HalfCircleSpinnerComponent() {
        this.animationDuration = 1000;
        this.size = 60;
        this.color = '#fff';
    }
    Object.defineProperty(HalfCircleSpinnerComponent.prototype, "spinnerStyle", {
        /**
         * @return {?}
         */
        get: function () {
            return {
                height: this.size + "px",
                width: this.size + "px",
            };
        },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(HalfCircleSpinnerComponent.prototype, "circleStyle", {
        /**
         * @return {?}
         */
        get: function () {
            return {
                borderWidth: this.size / 10 + "px",
                animationDuration: this.animationDuration + "ms"
            };
        },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(HalfCircleSpinnerComponent.prototype, "circle1Style", {
        /**
         * @return {?}
         */
        get: function () {
            return Object.assign({}, this.circleStyle, { borderTopColor: this.color });
        },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(HalfCircleSpinnerComponent.prototype, "circle2Style", {
        /**
         * @return {?}
         */
        get: function () {
            return Object.assign({}, this.circleStyle, { borderBottomColor: this.color });
        },
        enumerable: true,
        configurable: true
    });
    /**
     * @return {?}
     */
    HalfCircleSpinnerComponent.prototype.ngOnInit = function () {
    };
    return HalfCircleSpinnerComponent;
}());
HalfCircleSpinnerComponent.decorators = [
    { type: Component, args: [{
                selector: 'app-half-circle-spinner',
                template: "\n    <div class=\"half-circle-spinner\" [ngStyle]=\"spinnerStyle\">\n      <div class=\"circle circle-1\" [ngStyle]=\"circle1Style\"></div>\n      <div class=\"circle circle-2\" [ngStyle]=\"circle2Style\"></div>\n    </div>\n  ",
                styles: ["\n    .half-circle-spinner, .half-circle-spinner * {\n      -webkit-box-sizing: border-box;\n              box-sizing: border-box;\n    }\n    .half-circle-spinner {\n      width: 60px;\n      height: 60px;\n      border-radius: 100%;\n      position: relative;\n    }\n    .half-circle-spinner .circle {\n      content: \"\";\n      position: absolute;\n      width: 100%;\n      height: 100%;\n      border-radius: 100%;\n      border: calc(60px / 10) solid transparent;\n    }\n    .half-circle-spinner .circle.circle-1 {\n      border-top-color: #ff1d5e;\n      -webkit-animation: half-circle-spinner-animation 1s infinite;\n              animation: half-circle-spinner-animation 1s infinite;\n    }\n    .half-circle-spinner .circle.circle-2 {\n      border-bottom-color: #ff1d5e;\n      -webkit-animation: half-circle-spinner-animation 1s infinite alternate;\n              animation: half-circle-spinner-animation 1s infinite alternate;\n    }\n    @-webkit-keyframes half-circle-spinner-animation {\n      0% {\n        -webkit-transform: rotate(0deg);\n                transform: rotate(0deg);\n      }\n      100%{\n        -webkit-transform: rotate(360deg);\n                transform: rotate(360deg);\n      }\n    }\n    @keyframes half-circle-spinner-animation {\n      0% {\n        -webkit-transform: rotate(0deg);\n                transform: rotate(0deg);\n      }\n      100%{\n        -webkit-transform: rotate(360deg);\n                transform: rotate(360deg);\n      }\n    }\n  "]
            },] },
];
/**
 * @nocollapse
 */
HalfCircleSpinnerComponent.ctorParameters = function () { return []; };
HalfCircleSpinnerComponent.propDecorators = {
    'animationDuration': [{ type: Input },],
    'size': [{ type: Input },],
    'color': [{ type: Input },],
};
var HalfCircleSpinnerModule = /** @class */ (function () {
    function HalfCircleSpinnerModule() {
    }
    return HalfCircleSpinnerModule;
}());
HalfCircleSpinnerModule.decorators = [
    { type: NgModule, args: [{
                imports: [CommonModule],
                exports: [HalfCircleSpinnerComponent],
                declarations: [HalfCircleSpinnerComponent],
                providers: [],
            },] },
];
/**
 * @nocollapse
 */
HalfCircleSpinnerModule.ctorParameters = function () { return []; };
var HollowDotsSpinnerComponent = /** @class */ (function () {
    function HollowDotsSpinnerComponent() {
        this.animationDuration = 1000;
        this.dotSize = 15;
        this.dotsNum = 3;
        this.color = '#fff';
    }
    Object.defineProperty(HollowDotsSpinnerComponent.prototype, "horizontalMargin", {
        /**
         * @return {?}
         */
        get: function () {
            return this.dotSize / 2;
        },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(HollowDotsSpinnerComponent.prototype, "spinnerStyle", {
        /**
         * @return {?}
         */
        get: function () {
            return {
                height: this.dotSize + "px",
                width: this.dotSize + "px",
            };
        },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(HollowDotsSpinnerComponent.prototype, "dotStyle", {
        /**
         * @return {?}
         */
        get: function () {
            return {
                width: this.dotSize + "px",
                height: this.dotSize + "px",
                margin: "0 " + this.horizontalMargin + "px",
                borderWidth: this.dotSize / 5 + "px",
                borderColor: this.color,
                animationDuration: this.animationDuration + "ms"
            };
        },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(HollowDotsSpinnerComponent.prototype, "dotsStyles", {
        /**
         * @return {?}
         */
        get: function () {
            var /** @type {?} */ dotsStyles = [];
            var /** @type {?} */ delayModifier = 0.3;
            var /** @type {?} */ basicDelay = 1000;
            for (var /** @type {?} */ i = 1; i <= this.dotsNum; i++) {
                var /** @type {?} */ style = Object.assign({}, this.dotStyle, { animationDelay: basicDelay * i * delayModifier + "ms" });
                dotsStyles.push(style);
            }
            return dotsStyles;
        },
        enumerable: true,
        configurable: true
    });
    /**
     * @return {?}
     */
    HollowDotsSpinnerComponent.prototype.ngOnInit = function () {
    };
    return HollowDotsSpinnerComponent;
}());
HollowDotsSpinnerComponent.decorators = [
    { type: Component, args: [{
                selector: 'app-hollow-dots-spinner',
                template: "\n    <div class=\"hollow-dots-spinner\" [ngStyle]=\"spinnerStyle\">\n      <div class=\"dot\" *ngFor=\"let ds of dotsStyles\" [ngStyle]=\"ds\"></div>\n    </div>\n  ",
                styles: ["\n\n    .hollow-dots-spinner, .hollow-dots-spinner * {\n      -webkit-box-sizing: border-box;\n              box-sizing: border-box;\n    }\n    .hollow-dots-spinner {\n      height: 15px;\n      width: calc(30px * 3);\n    }\n    .hollow-dots-spinner .dot {\n      width: 15px;\n      height: 15px;\n      margin: 0 calc(15px / 2);\n      border: calc(15px / 5) solid #ff1d5e;\n      border-radius: 50%;\n      float: left;\n      -webkit-transform: scale(0);\n              transform: scale(0);\n      -webkit-animation: hollow-dots-spinner-animation 1000ms ease infinite 0ms;\n              animation: hollow-dots-spinner-animation 1000ms ease infinite 0ms;\n    }\n    .hollow-dots-spinner .dot:nth-child(1) {\n      -webkit-animation-delay: calc(300ms * 1);\n              animation-delay: calc(300ms * 1);\n    }\n    .hollow-dots-spinner .dot:nth-child(2) {\n      -webkit-animation-delay: calc(300ms * 2);\n              animation-delay: calc(300ms * 2);\n    }\n    .hollow-dots-spinner .dot:nth-child(3) {\n      -webkit-animation-delay: calc(300ms * 3);\n              animation-delay: calc(300ms * 3);\n    }\n    @-webkit-keyframes hollow-dots-spinner-animation {\n      50% {\n        -webkit-transform: scale(1);\n                transform: scale(1);\n        opacity: 1;\n      }\n      100% {\n        opacity: 0;\n      }\n    }\n    @keyframes hollow-dots-spinner-animation {\n      50% {\n        -webkit-transform: scale(1);\n                transform: scale(1);\n        opacity: 1;\n      }\n      100% {\n        opacity: 0;\n      }\n    }\n  "]
            },] },
];
/**
 * @nocollapse
 */
HollowDotsSpinnerComponent.ctorParameters = function () { return []; };
HollowDotsSpinnerComponent.propDecorators = {
    'animationDuration': [{ type: Input },],
    'dotSize': [{ type: Input },],
    'dotsNum': [{ type: Input },],
    'color': [{ type: Input },],
};
var HollowDotsSpinnerModule = /** @class */ (function () {
    function HollowDotsSpinnerModule() {
    }
    return HollowDotsSpinnerModule;
}());
HollowDotsSpinnerModule.decorators = [
    { type: NgModule, args: [{
                imports: [CommonModule],
                exports: [HollowDotsSpinnerComponent],
                declarations: [HollowDotsSpinnerComponent],
                providers: [],
            },] },
];
/**
 * @nocollapse
 */
HollowDotsSpinnerModule.ctorParameters = function () { return []; };
var IntersectingCirclesSpinnerComponent = /** @class */ (function () {
    function IntersectingCirclesSpinnerComponent() {
        this.animationDuration = 1200;
        this.size = 70;
        this.color = '#fff';
    }
    Object.defineProperty(IntersectingCirclesSpinnerComponent.prototype, "circleSize", {
        /**
         * @return {?}
         */
        get: function () {
            return this.size / 2;
        },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(IntersectingCirclesSpinnerComponent.prototype, "spinnerStyle", {
        /**
         * @return {?}
         */
        get: function () {
            return {
                width: this.size + "px",
                height: this.size + "px"
            };
        },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(IntersectingCirclesSpinnerComponent.prototype, "spinnerBlockStyle", {
        /**
         * @return {?}
         */
        get: function () {
            return {
                animationDuration: this.animationDuration + "ms",
                width: this.circleSize + "px",
                height: this.circleSize + "px"
            };
        },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(IntersectingCirclesSpinnerComponent.prototype, "circleStyle", {
        /**
         * @return {?}
         */
        get: function () {
            return {
                borderColor: this.color
            };
        },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(IntersectingCirclesSpinnerComponent.prototype, "circleStyles", {
        /**
         * @return {?}
         */
        get: function () {
            var _this = this;
            var /** @type {?} */ circlesPositions = [
                { top: 0, left: 0 },
                { left: this.circleSize * -0.36 + "px", top: this.circleSize * 0.2 + "px" },
                { left: this.circleSize * -0.36 + "px", top: this.circleSize * -0.2 + "px" },
                { left: 0, top: this.circleSize * -0.36 + "px" },
                { left: this.circleSize * 0.36 + "px", top: this.circleSize * -0.2 + "px" },
                { left: this.circleSize * 0.36 + "px", top: this.circleSize * 0.2 + "px" },
                { left: 0, top: this.circleSize * 0.36 + "px" }
            ];
            return circlesPositions.map(function (cp) {
                return Object.assign({ cp: cp }, _this.circleStyle);
            });
        },
        enumerable: true,
        configurable: true
    });
    /**
     * @return {?}
     */
    IntersectingCirclesSpinnerComponent.prototype.ngOnInit = function () {
    };
    return IntersectingCirclesSpinnerComponent;
}());
IntersectingCirclesSpinnerComponent.decorators = [
    { type: Component, args: [{
                selector: 'app-intersecting-circles-spinner',
                template: "\n    <div class=\"intersecting-circles-spinner\" [ngStyle]=\"spinnerStyle\">\n      <div class=\"spinnerBlock\" [ngStyle]=\"spinnerBlockStyle\">\n        <span class=\"circle\" *ngFor=\"let cs of circleStyles\" [ngStyle]=\"cs\"></span>\n      </div>\n    </div>\n  ",
                styles: ["\n    .intersecting-circles-spinner, .intersecting-circles-spinner * {\n      -webkit-box-sizing: border-box;\n              box-sizing: border-box;\n    }\n    .intersecting-circles-spinner {\n      height: 70px;\n      width: 70px;\n      position: relative;\n      display: -webkit-box;\n      display: -ms-flexbox;\n      display: flex;\n      -webkit-box-orient: horizontal;\n      -webkit-box-direction: normal;\n          -ms-flex-direction: row;\n              flex-direction: row;\n      -webkit-box-pack: center;\n          -ms-flex-pack: center;\n              justify-content: center;\n      -webkit-box-align: center;\n          -ms-flex-align: center;\n              align-items: center;\n    }\n    .intersecting-circles-spinner .spinnerBlock {\n      -webkit-animation: intersecting-circles-spinners-animation 1200ms linear infinite;\n              animation: intersecting-circles-spinners-animation 1200ms linear infinite;\n      -webkit-transform-origin: center;\n              transform-origin: center;\n      display: block;\n      height: 35px;\n      width: 35px;\n    }\n    .intersecting-circles-spinner .circle {\n      display: block;\n      border: 2px solid #ff1d5e;\n      border-radius: 50%;\n      height: 100%;\n      width: 100%;\n      position: absolute;\n      left: 0;\n      top: 0;\n    }\n    .intersecting-circles-spinner .circle:nth-child(1) {\n      left: 0;\n      top: 0;\n    }\n    .intersecting-circles-spinner .circle:nth-child(2) {\n      left: calc(35px * -0.36);\n      top: calc(35px * 0.2);\n    }\n    .intersecting-circles-spinner .circle:nth-child(3) {\n      left: calc(35px * -0.36);\n      top: calc(35px * -0.2);\n    }\n    .intersecting-circles-spinner .circle:nth-child(4) {\n      left: 0;\n      top: calc(35px * -0.36);\n    }\n    .intersecting-circles-spinner .circle:nth-child(5) {\n      left: calc(35px * 0.36);\n      top: calc(35px * -0.2);\n    }\n    .intersecting-circles-spinner .circle:nth-child(6) {\n      left: calc(35px * 0.36);\n      top: calc(35px * 0.2);\n    }\n    .intersecting-circles-spinner .circle:nth-child(7) {\n      left: 0;\n      top: calc(35px * 0.36);\n    }\n    @-webkit-keyframes intersecting-circles-spinners-animation {\n      from { -webkit-transform: rotate(0deg); transform: rotate(0deg); }\n      to { -webkit-transform: rotate(360deg); transform: rotate(360deg); }\n    }\n    @keyframes intersecting-circles-spinners-animation {\n      from { -webkit-transform: rotate(0deg); transform: rotate(0deg); }\n      to { -webkit-transform: rotate(360deg); transform: rotate(360deg); }\n    }\n  "]
            },] },
];
/**
 * @nocollapse
 */
IntersectingCirclesSpinnerComponent.ctorParameters = function () { return []; };
IntersectingCirclesSpinnerComponent.propDecorators = {
    'animationDuration': [{ type: Input },],
    'size': [{ type: Input },],
    'color': [{ type: Input },],
};
var IntersectingCirclesSpinnerModule = /** @class */ (function () {
    function IntersectingCirclesSpinnerModule() {
    }
    return IntersectingCirclesSpinnerModule;
}());
IntersectingCirclesSpinnerModule.decorators = [
    { type: NgModule, args: [{
                imports: [CommonModule],
                exports: [IntersectingCirclesSpinnerComponent],
                declarations: [IntersectingCirclesSpinnerComponent],
                providers: [],
            },] },
];
/**
 * @nocollapse
 */
IntersectingCirclesSpinnerModule.ctorParameters = function () { return []; };
var LoopingRhumbusesSpinnerComponent = /** @class */ (function () {
    function LoopingRhumbusesSpinnerComponent() {
        this.animationDuration = 2500;
        this.rhombusSize = 15;
        this.color = '#fff';
        this.rhombusesNum = 3;
    }
    Object.defineProperty(LoopingRhumbusesSpinnerComponent.prototype, "spinnerStyle", {
        /**
         * @return {?}
         */
        get: function () {
            return {
                height: this.rhombusSize + "px",
                width: this.rhombusSize * 4 + "px"
            };
        },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(LoopingRhumbusesSpinnerComponent.prototype, "rhombusStyle", {
        /**
         * @return {?}
         */
        get: function () {
            return {
                height: this.rhombusSize + "px",
                width: this.rhombusSize + "px",
                background: this.color,
                animationDuration: this.animationDuration + "ms",
                left: this.rhombusSize * 4 + "px"
            };
        },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(LoopingRhumbusesSpinnerComponent.prototype, "rhombusesStyles", {
        /**
         * @return {?}
         */
        get: function () {
            var /** @type {?} */ rhombusesStyles = [];
            var /** @type {?} */ delay = -this.animationDuration / 1.5;
            for (var /** @type {?} */ i = 1; i <= this.rhombusesNum; i++) {
                var /** @type {?} */ style = Object.assign({
                    animationDelay: i * delay + "ms"
                }, this.rhombusStyle);
                rhombusesStyles.push(style);
            }
            return rhombusesStyles;
        },
        enumerable: true,
        configurable: true
    });
    /**
     * @return {?}
     */
    LoopingRhumbusesSpinnerComponent.prototype.ngOnInit = function () {
    };
    return LoopingRhumbusesSpinnerComponent;
}());
LoopingRhumbusesSpinnerComponent.decorators = [
    { type: Component, args: [{
                selector: 'app-looping-rhombuses-spinner',
                template: "\n    <div class=\"looping-rhombuses-spinner\" [ngStyle]=\"spinnerStyle\">\n      <div class=\"rhombus\"\n           *ngFor=\"let rs of rhombusesStyles\"\n           [ngStyle]=\"rs\">\n      </div>\n    </div>\n  ",
                styles: ["\n    .looping-rhombuses-spinner, .looping-rhombuses-spinner * {\n      -webkit-box-sizing: border-box;\n              box-sizing: border-box;\n    }\n    .looping-rhombuses-spinner {\n      width: calc(15px * 4);\n      height: 15px;\n      position: relative;\n    }\n    .looping-rhombuses-spinner .rhombus {\n      height: 15px;\n      width: 15px;\n      background: #ff1d5e;\n      left: calc(15px * 4);\n      position: absolute;\n      margin: 0 auto;\n      border-radius: 2px;\n      -webkit-transform: translateY(0) rotate(45deg) scale(0);\n              transform: translateY(0) rotate(45deg) scale(0);\n      -webkit-animation: looping-rhombuses-spinner-animation 2500ms linear infinite;\n              animation: looping-rhombuses-spinner-animation 2500ms linear infinite;\n    }\n    .looping-rhombuses-spinner .rhombus:nth-child(1) {\n      -webkit-animation-delay: calc(2500ms * 1 / -1.5);\n              animation-delay: calc(2500ms * 1 / -1.5);\n    }\n    .looping-rhombuses-spinner .rhombus:nth-child(2) {\n      -webkit-animation-delay: calc(2500ms * 2 / -1.5);\n              animation-delay: calc(2500ms * 2 / -1.5);\n    }\n    .looping-rhombuses-spinner .rhombus:nth-child(3) {\n      -webkit-animation-delay: calc(2500ms * 3 / -1.5);\n              animation-delay: calc(2500ms * 3 / -1.5);\n    }\n    @-webkit-keyframes looping-rhombuses-spinner-animation {\n      0% {\n        -webkit-transform: translateX(0) rotate(45deg) scale(0);\n                transform: translateX(0) rotate(45deg) scale(0);\n      }\n      50% {\n        -webkit-transform: translateX(-233%) rotate(45deg) scale(1);\n                transform: translateX(-233%) rotate(45deg) scale(1);\n      }\n      100% {\n        -webkit-transform: translateX(-466%) rotate(45deg) scale(0);\n                transform: translateX(-466%) rotate(45deg) scale(0);\n      }\n    }\n    @keyframes looping-rhombuses-spinner-animation {\n      0% {\n        -webkit-transform: translateX(0) rotate(45deg) scale(0);\n                transform: translateX(0) rotate(45deg) scale(0);\n      }\n      50% {\n        -webkit-transform: translateX(-233%) rotate(45deg) scale(1);\n                transform: translateX(-233%) rotate(45deg) scale(1);\n      }\n      100% {\n        -webkit-transform: translateX(-466%) rotate(45deg) scale(0);\n                transform: translateX(-466%) rotate(45deg) scale(0);\n      }\n    }\n  "]
            },] },
];
/**
 * @nocollapse
 */
LoopingRhumbusesSpinnerComponent.ctorParameters = function () { return []; };
LoopingRhumbusesSpinnerComponent.propDecorators = {
    'animationDuration': [{ type: Input },],
    'rhombusSize': [{ type: Input },],
    'color': [{ type: Input },],
};
var LoopingRhumbusesSpinnerModule = /** @class */ (function () {
    function LoopingRhumbusesSpinnerModule() {
    }
    return LoopingRhumbusesSpinnerModule;
}());
LoopingRhumbusesSpinnerModule.decorators = [
    { type: NgModule, args: [{
                imports: [CommonModule],
                exports: [LoopingRhumbusesSpinnerComponent],
                declarations: [LoopingRhumbusesSpinnerComponent],
                providers: [],
            },] },
];
/**
 * @nocollapse
 */
LoopingRhumbusesSpinnerModule.ctorParameters = function () { return []; };
var OrbitSpinnerComponent = /** @class */ (function () {
    function OrbitSpinnerComponent() {
        this.animationDuration = 1000;
        this.size = 50;
        this.color = '#fff';
    }
    Object.defineProperty(OrbitSpinnerComponent.prototype, "spinnerStyle", {
        /**
         * @return {?}
         */
        get: function () {
            return {
                height: this.size + "px",
                width: this.size + "px"
            };
        },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(OrbitSpinnerComponent.prototype, "orbitStyle", {
        /**
         * @return {?}
         */
        get: function () {
            return {
                borderColor: this.color,
                animationDuration: this.animationDuration + "ms"
            };
        },
        enumerable: true,
        configurable: true
    });
    /**
     * @return {?}
     */
    OrbitSpinnerComponent.prototype.ngOnInit = function () {
    };
    return OrbitSpinnerComponent;
}());
OrbitSpinnerComponent.decorators = [
    { type: Component, args: [{
                selector: 'app-orbit-spinner',
                template: "\n    <div class=\"orbit-spinner\" [ngStyle]=\"spinnerStyle\">\n      <div class=\"orbit one\" [ngStyle]=\"orbitStyle\"></div>\n      <div class=\"orbit two\" [ngStyle]=\"orbitStyle\"></div>\n      <div class=\"orbit three\" [ngStyle]=\"orbitStyle\"></div>\n    </div>\n  ",
                styles: ["\n    .orbit-spinner, .orbit-spinner * {\n      -webkit-box-sizing: border-box;\n              box-sizing: border-box;\n    }\n    .orbit-spinner {\n      height: 55px;\n      width: 55px;\n      border-radius: 50%;\n      -webkit-perspective: 800px;\n              perspective: 800px;\n    }\n    .orbit-spinner .orbit {\n      position: absolute;\n      -webkit-box-sizing: border-box;\n              box-sizing: border-box;\n      width: 100%;\n      height: 100%;\n      border-radius: 50%;\n    }\n    .orbit-spinner .orbit:nth-child(1) {\n      left: 0;\n      top: 0;\n      -webkit-animation: orbit-spinner-orbit-one-animation 1200ms linear infinite;\n              animation: orbit-spinner-orbit-one-animation 1200ms linear infinite;\n      border-bottom: 3px solid #ff1d5e;\n    }\n    .orbit-spinner .orbit:nth-child(2) {\n      right: 0;\n      top: 0;\n      -webkit-animation: orbit-spinner-orbit-two-animation 1200ms linear infinite;\n              animation: orbit-spinner-orbit-two-animation 1200ms linear infinite;\n      border-right: 3px solid #ff1d5e;\n    }\n    .orbit-spinner .orbit:nth-child(3) {\n      right: 0;\n      bottom: 0;\n      -webkit-animation: orbit-spinner-orbit-three-animation 1200ms linear infinite;\n              animation: orbit-spinner-orbit-three-animation 1200ms linear infinite;\n      border-top: 3px solid #ff1d5e;\n    }\n    @-webkit-keyframes orbit-spinner-orbit-one-animation {\n      0%{\n        -webkit-transform: rotateX(35deg) rotateY(-45deg) rotateZ(0deg);\n                transform: rotateX(35deg) rotateY(-45deg) rotateZ(0deg);\n      }\n      100% {\n        -webkit-transform: rotateX(35deg) rotateY(-45deg) rotateZ(360deg);\n                transform: rotateX(35deg) rotateY(-45deg) rotateZ(360deg);\n      }\n    }\n    @keyframes orbit-spinner-orbit-one-animation {\n      0%{\n        -webkit-transform: rotateX(35deg) rotateY(-45deg) rotateZ(0deg);\n                transform: rotateX(35deg) rotateY(-45deg) rotateZ(0deg);\n      }\n      100% {\n        -webkit-transform: rotateX(35deg) rotateY(-45deg) rotateZ(360deg);\n                transform: rotateX(35deg) rotateY(-45deg) rotateZ(360deg);\n      }\n    }\n    @-webkit-keyframes orbit-spinner-orbit-two-animation {\n      0% {\n        -webkit-transform: rotateX(50deg) rotateY(10deg) rotateZ(0deg);\n                transform: rotateX(50deg) rotateY(10deg) rotateZ(0deg);\n      }\n      100% {\n        -webkit-transform: rotateX(50deg) rotateY(10deg) rotateZ(360deg);\n                transform: rotateX(50deg) rotateY(10deg) rotateZ(360deg);\n      }\n    }\n    @keyframes orbit-spinner-orbit-two-animation {\n      0% {\n        -webkit-transform: rotateX(50deg) rotateY(10deg) rotateZ(0deg);\n                transform: rotateX(50deg) rotateY(10deg) rotateZ(0deg);\n      }\n      100% {\n        -webkit-transform: rotateX(50deg) rotateY(10deg) rotateZ(360deg);\n                transform: rotateX(50deg) rotateY(10deg) rotateZ(360deg);\n      }\n    }\n    @-webkit-keyframes orbit-spinner-orbit-three-animation {\n      0% {\n        -webkit-transform: rotateX(35deg) rotateY(55deg) rotateZ(0deg);\n                transform: rotateX(35deg) rotateY(55deg) rotateZ(0deg);\n      }\n      100% {\n        -webkit-transform: rotateX(35deg) rotateY(55deg) rotateZ(360deg);\n                transform: rotateX(35deg) rotateY(55deg) rotateZ(360deg);\n      }\n    }\n    @keyframes orbit-spinner-orbit-three-animation {\n      0% {\n        -webkit-transform: rotateX(35deg) rotateY(55deg) rotateZ(0deg);\n                transform: rotateX(35deg) rotateY(55deg) rotateZ(0deg);\n      }\n      100% {\n        -webkit-transform: rotateX(35deg) rotateY(55deg) rotateZ(360deg);\n                transform: rotateX(35deg) rotateY(55deg) rotateZ(360deg);\n      }\n    }\n  "]
            },] },
];
/**
 * @nocollapse
 */
OrbitSpinnerComponent.ctorParameters = function () { return []; };
OrbitSpinnerComponent.propDecorators = {
    'animationDuration': [{ type: Input },],
    'size': [{ type: Input },],
    'color': [{ type: Input },],
};
var OrbitSpinnerModule = /** @class */ (function () {
    function OrbitSpinnerModule() {
    }
    return OrbitSpinnerModule;
}());
OrbitSpinnerModule.decorators = [
    { type: NgModule, args: [{
                imports: [CommonModule],
                exports: [OrbitSpinnerComponent],
                declarations: [OrbitSpinnerComponent],
                providers: [],
            },] },
];
/**
 * @nocollapse
 */
OrbitSpinnerModule.ctorParameters = function () { return []; };
var PixelSpinnerComponent = /** @class */ (function () {
    function PixelSpinnerComponent() {
        this.animationDuration = 1000;
        this.size = 50;
        this.color = '#fff';
        this.animationBaseName = 'pixel-spinner-animation';
        this.currentAnimationName = '';
    }
    Object.defineProperty(PixelSpinnerComponent.prototype, "pixelSize", {
        /**
         * @return {?}
         */
        get: function () {
            return this.size / 7;
        },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(PixelSpinnerComponent.prototype, "spinnerStyle", {
        /**
         * @return {?}
         */
        get: function () {
            return {
                width: this.size + "px",
                height: this.size + "px"
            };
        },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(PixelSpinnerComponent.prototype, "spinnerInnerStyle", {
        /**
         * @return {?}
         */
        get: function () {
            return {
                animationDuration: this.animationDuration + "ms",
                animationName: this.currentAnimationName,
                width: this.pixelSize + "px",
                height: this.pixelSize + "px",
                backgroundColor: this.color,
                color: this.color,
                boxShadow: "\n                        " + this.pixelSize * 1.5 + "px " + this.pixelSize * 1.5 + "px 0 0,\n                        " + this.pixelSize * -1.5 + "px " + this.pixelSize * -1.5 + "px 0 0,\n                        " + this.pixelSize * 1.5 + "px " + this.pixelSize * -1.5 + "px 0 0,\n                        " + this.pixelSize * -1.5 + "px " + this.pixelSize * 1.5 + "px 0 0,\n                        0 " + this.pixelSize * 1.5 + "px 0 0,\n                        " + this.pixelSize * 1.5 + "px 0 0 0,\n                        " + this.pixelSize * -1.5 + "px 0 0 0,\n                        0 " + this.pixelSize * -1.5 + "px 0 0\n                      "
            };
        },
        enumerable: true,
        configurable: true
    });
    /**
     * @return {?}
     */
    PixelSpinnerComponent.prototype.ngOnInit = function () {
    };
    return PixelSpinnerComponent;
}());
PixelSpinnerComponent.decorators = [
    { type: Component, args: [{
                selector: 'app-pixel-spinner',
                template: "\n    <div class=\"pixel-spinner\" [ngStyle]=\"spinnerStyle\">\n      <div class=\"pixel-spinner-inner\" [ngStyle]=\"spinnerInnerStyle\"></div>\n    </div>\n  ",
                styles: ["\n    .pixel-spinner, .pixel-spinner * {\n      -webkit-box-sizing: border-box;\n              box-sizing: border-box;\n    }\n\n    .pixel-spinner {\n      height: 70px;\n      width: 70px;\n      display: -webkit-box;\n      display: -ms-flexbox;\n      display: flex;\n      -webkit-box-orient: horizontal;\n      -webkit-box-direction: normal;\n          -ms-flex-direction: row;\n              flex-direction: row;\n      -webkit-box-pack: center;\n          -ms-flex-pack: center;\n              justify-content: center;\n      -webkit-box-align: center;\n          -ms-flex-align: center;\n              align-items: center;\n    }\n\n    .pixel-spinner .pixel-spinner-inner {\n      width: calc(70px / 7);\n      height: calc(70px / 7);\n      background-color: #ff1d5e;\n      color: #ff1d5e;\n      -webkit-box-shadow: 15px 15px 0 0,\n      -15px -15px 0 0,\n      15px -15px 0 0,\n      -15px 15px 0 0,\n      0 15px 0 0,\n      15px 0 0 0,\n      -15px 0 0 0,\n      0 -15px 0 0;\n              box-shadow: 15px 15px 0 0,\n      -15px -15px 0 0,\n      15px -15px 0 0,\n      -15px 15px 0 0,\n      0 15px 0 0,\n      15px 0 0 0,\n      -15px 0 0 0,\n      0 -15px 0 0;\n      -webkit-animation: pixel-spinner-animation 2000ms linear infinite;\n              animation: pixel-spinner-animation 2000ms linear infinite;\n    }\n\n    @-webkit-keyframes pixel-spinner-animation {\n      50% {\n        -webkit-box-shadow: 20px 20px 0 0,\n        -20px -20px 0 0,\n        20px -20px 0 0,\n        -20px 20px 0 0,\n        0 10px 0 0,\n        10px 0 0 0,\n        -10px 0 0 0,\n        0 -10px 0 0;\n                box-shadow: 20px 20px 0 0,\n        -20px -20px 0 0,\n        20px -20px 0 0,\n        -20px 20px 0 0,\n        0 10px 0 0,\n        10px 0 0 0,\n        -10px 0 0 0,\n        0 -10px 0 0;\n      }\n      75% {\n        -webkit-box-shadow: 20px 20px 0 0,\n        -20px -20px 0 0,\n        20px -20px 0 0,\n        -20px 20px 0 0,\n        0 10px 0 0,\n        10px 0 0 0,\n        -10px 0 0 0,\n        0 -10px 0 0;\n                box-shadow: 20px 20px 0 0,\n        -20px -20px 0 0,\n        20px -20px 0 0,\n        -20px 20px 0 0,\n        0 10px 0 0,\n        10px 0 0 0,\n        -10px 0 0 0,\n        0 -10px 0 0;\n      }\n      100% {\n        -webkit-transform: rotate(360deg);\n                transform: rotate(360deg);\n      }\n    }\n\n    @keyframes pixel-spinner-animation {\n      50% {\n        -webkit-box-shadow: 20px 20px 0 0,\n        -20px -20px 0 0,\n        20px -20px 0 0,\n        -20px 20px 0 0,\n        0 10px 0 0,\n        10px 0 0 0,\n        -10px 0 0 0,\n        0 -10px 0 0;\n                box-shadow: 20px 20px 0 0,\n        -20px -20px 0 0,\n        20px -20px 0 0,\n        -20px 20px 0 0,\n        0 10px 0 0,\n        10px 0 0 0,\n        -10px 0 0 0,\n        0 -10px 0 0;\n      }\n      75% {\n        -webkit-box-shadow: 20px 20px 0 0,\n        -20px -20px 0 0,\n        20px -20px 0 0,\n        -20px 20px 0 0,\n        0 10px 0 0,\n        10px 0 0 0,\n        -10px 0 0 0,\n        0 -10px 0 0;\n                box-shadow: 20px 20px 0 0,\n        -20px -20px 0 0,\n        20px -20px 0 0,\n        -20px 20px 0 0,\n        0 10px 0 0,\n        10px 0 0 0,\n        -10px 0 0 0,\n        0 -10px 0 0;\n      }\n      100% {\n        -webkit-transform: rotate(360deg);\n                transform: rotate(360deg);\n      }\n    }\n  "]
            },] },
];
/**
 * @nocollapse
 */
PixelSpinnerComponent.ctorParameters = function () { return []; };
PixelSpinnerComponent.propDecorators = {
    'animationDuration': [{ type: Input },],
    'size': [{ type: Input },],
    'color': [{ type: Input },],
};
var PixelSpinnerModule = /** @class */ (function () {
    function PixelSpinnerModule() {
    }
    return PixelSpinnerModule;
}());
PixelSpinnerModule.decorators = [
    { type: NgModule, args: [{
                imports: [CommonModule],
                exports: [PixelSpinnerComponent],
                declarations: [PixelSpinnerComponent],
                providers: [],
            },] },
];
/**
 * @nocollapse
 */
PixelSpinnerModule.ctorParameters = function () { return []; };
var RadarSpinnerComponent = /** @class */ (function () {
    function RadarSpinnerComponent() {
        this.size = 110;
        this.animationDuration = 2000;
        this.color = '#fff';
        this.circlesNum = 4;
    }
    Object.defineProperty(RadarSpinnerComponent.prototype, "borderWidth", {
        /**
         * @return {?}
         */
        get: function () {
            return this.size * 5 / 110;
        },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(RadarSpinnerComponent.prototype, "spinnerStyle", {
        /**
         * @return {?}
         */
        get: function () {
            return {
                height: this.size + "px",
                width: this.size + "px"
            };
        },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(RadarSpinnerComponent.prototype, "circleStyle", {
        /**
         * @return {?}
         */
        get: function () {
            return {
                animationDuration: this.animationDuration + "ms"
            };
        },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(RadarSpinnerComponent.prototype, "circleInnerContainerStyle", {
        /**
         * @return {?}
         */
        get: function () {
            return {
                borderWidth: this.borderWidth + "px"
            };
        },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(RadarSpinnerComponent.prototype, "circleInnerStyle", {
        /**
         * @return {?}
         */
        get: function () {
            return {
                borderLeftColor: this.color,
                borderRightColor: this.color,
                borderWidth: this.borderWidth + "px"
            };
        },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(RadarSpinnerComponent.prototype, "circlesStyles", {
        /**
         * @return {?}
         */
        get: function () {
            var /** @type {?} */ circlesStyles = [];
            var /** @type {?} */ delay = this.animationDuration * 0.15;
            for (var /** @type {?} */ i = 0; i < this.circlesNum; i++) {
                circlesStyles.push(Object.assign({
                    padding: this.borderWidth * 2 * i + "px",
                    animationDelay: (i === this.circlesNum - 1 ? 0 : delay) + "ms"
                }, this.circleStyle));
            }
            return circlesStyles;
        },
        enumerable: true,
        configurable: true
    });
    /**
     * @return {?}
     */
    RadarSpinnerComponent.prototype.ngOnInit = function () {
    };
    return RadarSpinnerComponent;
}());
RadarSpinnerComponent.decorators = [
    { type: Component, args: [{
                selector: 'app-radar-spinner',
                template: "\n    <div class=\"radar-spinner\" [ngStyle]=\"spinnerStyle\">\n      <div class=\"circle\" *ngFor=\"let cs of circlesStyles\" [ngStyle]=\"cs\">\n        <div class=\"circle-inner-container\" [ngStyle]=\"circleInnerContainerStyle\">\n          <div class=\"circle-inner\" [ngStyle]=\"circleInnerStyle\"></div>\n        </div>\n      </div>\n    </div>\n  ",
                styles: ["\n    .radar-spinner, .radar-spinner * {\n      -webkit-box-sizing: border-box;\n              box-sizing: border-box;\n    }\n    .radar-spinner {\n      height: 60px;\n      width: 60px;\n      position: relative;\n    }\n    .radar-spinner .circle {\n      position: absolute;\n      height: 100%;\n      width: 100%;\n      top: 0;\n      left: 0;\n      -webkit-animation: radar-spinner-animation 2s infinite;\n              animation: radar-spinner-animation 2s infinite;\n    }\n    .radar-spinner .circle:nth-child(1) {\n      padding: calc(60px * 5 * 2 * 0 / 110);\n      -webkit-animation-delay: 300ms;\n              animation-delay: 300ms;\n    }\n    .radar-spinner .circle:nth-child(2) {\n      padding: calc(60px * 5 * 2 * 1 / 110);\n      -webkit-animation-delay: 300ms;\n              animation-delay: 300ms;\n    }\n    .radar-spinner .circle:nth-child(3) {\n      padding: calc(60px * 5 * 2 * 2 / 110);\n      -webkit-animation-delay: 300ms;\n              animation-delay: 300ms;\n    }\n    .radar-spinner .circle:nth-child(4) {\n      padding: calc(60px * 5 * 2 * 3 / 110);\n      -webkit-animation-delay: 0ms;\n              animation-delay: 0ms;\n    }\n    .radar-spinner .circle-inner, .radar-spinner .circle-inner-container {\n      height: 100%;\n      width: 100%;\n      border-radius: 50%;\n      border: calc(60px * 5 / 110) solid transparent;\n    }\n    .radar-spinner .circle-inner {\n      border-left-color: #ff1d5e;\n      border-right-color: #ff1d5e;\n    }\n    @-webkit-keyframes radar-spinner-animation {\n      50% {\n        -webkit-transform: rotate(180deg);\n                transform: rotate(180deg);\n      }\n      100% {\n        -webkit-transform: rotate(0deg);\n                transform: rotate(0deg);\n      }\n    }\n    @keyframes radar-spinner-animation {\n      50% {\n        -webkit-transform: rotate(180deg);\n                transform: rotate(180deg);\n      }\n      100% {\n        -webkit-transform: rotate(0deg);\n                transform: rotate(0deg);\n      }\n    }\n  "]
            },] },
];
/**
 * @nocollapse
 */
RadarSpinnerComponent.ctorParameters = function () { return []; };
RadarSpinnerComponent.propDecorators = {
    'size': [{ type: Input },],
    'animationDuration': [{ type: Input },],
    'color': [{ type: Input },],
};
var RadarSpinnerModule = /** @class */ (function () {
    function RadarSpinnerModule() {
    }
    return RadarSpinnerModule;
}());
RadarSpinnerModule.decorators = [
    { type: NgModule, args: [{
                imports: [CommonModule],
                exports: [RadarSpinnerComponent],
                declarations: [RadarSpinnerComponent],
                providers: [],
            },] },
];
/**
 * @nocollapse
 */
RadarSpinnerModule.ctorParameters = function () { return []; };
var ScalingSquaresSpinnerComponent = /** @class */ (function () {
    function ScalingSquaresSpinnerComponent() {
        this.size = 65;
        this.color = '#fff';
        this.animationDuration = 1250;
        this.squaresNum = 4;
    }
    Object.defineProperty(ScalingSquaresSpinnerComponent.prototype, "spinnerStyle", {
        /**
         * @return {?}
         */
        get: function () {
            return {
                height: this.size + "px",
                width: this.size + "px",
                animationDuration: this.animationDuration + "ms"
            };
        },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(ScalingSquaresSpinnerComponent.prototype, "squareStyle", {
        /**
         * @return {?}
         */
        get: function () {
            return {
                height: this.size * 0.25 / 1.3 + "px",
                width: this.size * 0.25 / 1.3 + "px",
                animationDuration: this.animationDuration + "ms",
                borderWidth: this.size * 0.04 / 1.3 + "px",
                borderColor: this.color
            };
        },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(ScalingSquaresSpinnerComponent.prototype, "squaresStyles", {
        /**
         * @return {?}
         */
        get: function () {
            var /** @type {?} */ squaresStyles = [];
            for (var /** @type {?} */ i = 1; i <= this.squaresNum; i++) {
                squaresStyles.push(Object.assign({}, this.squareStyle));
            }
            return squaresStyles;
        },
        enumerable: true,
        configurable: true
    });
    /**
     * @param {?} i
     * @return {?}
     */
    ScalingSquaresSpinnerComponent.prototype.childClass = function (i) {
        return "square-" + (i + 1);
    };
    /**
     * @return {?}
     */
    ScalingSquaresSpinnerComponent.prototype.ngOnInit = function () {
    };
    return ScalingSquaresSpinnerComponent;
}());
ScalingSquaresSpinnerComponent.decorators = [
    { type: Component, args: [{
                selector: 'app-scaling-squares-spinner',
                template: "\n    <div class=\"scaling-squares-spinner\" [ngStyle]=\"spinnerStyle\">\n      <div class=\"square\"\n           *ngFor=\"let ss of squaresStyles; let i = index;\"\n           [ngClass]=\"childClass(i)\"\n           [ngStyle]=\"ss\">\n      </div>\n    </div>\n  ",
                styles: ["\n    .scaling-squares-spinner, .scaling-squares-spinner * {\n      -webkit-box-sizing: border-box;\n              box-sizing: border-box;\n    }\n    .scaling-squares-spinner {\n      height: 65px;\n      width: 65px;\n      position: relative;\n      display: -webkit-box;\n      display: -ms-flexbox;\n      display: flex;\n      -webkit-box-orient: horizontal;\n      -webkit-box-direction: normal;\n          -ms-flex-direction: row;\n              flex-direction: row;\n      -webkit-box-align: center;\n          -ms-flex-align: center;\n              align-items: center;\n      -webkit-box-pack: center;\n          -ms-flex-pack: center;\n              justify-content: center;\n      -webkit-animation: scaling-squares-animation 1250ms;\n              animation: scaling-squares-animation 1250ms;\n      -webkit-animation-iteration-count: infinite;\n              animation-iteration-count: infinite;\n      -webkit-transform: rotate(0deg);\n              transform: rotate(0deg);\n    }\n    .scaling-squares-spinner .square {\n      height: calc(65px * 0.25 / 1.3);\n      width: calc(65px * 0.25 / 1.3);\n      margin-right: auto;\n      margin-left: auto;\n      border: calc(65px * 0.04 / 1.3) solid #ff1d5e;\n      position: absolute;\n      -webkit-animation-duration: 1250ms;\n              animation-duration: 1250ms;\n      -webkit-animation-iteration-count: infinite;\n              animation-iteration-count: infinite;\n    }\n    .scaling-squares-spinner .square:nth-child(1) {\n      -webkit-animation-name: scaling-squares-spinner-animation-child-1;\n              animation-name: scaling-squares-spinner-animation-child-1;\n    }\n    .scaling-squares-spinner .square:nth-child(2) {\n      -webkit-animation-name: scaling-squares-spinner-animation-child-2;\n              animation-name: scaling-squares-spinner-animation-child-2;\n    }\n    .scaling-squares-spinner .square:nth-child(3) {\n      -webkit-animation-name: scaling-squares-spinner-animation-child-3;\n              animation-name: scaling-squares-spinner-animation-child-3;\n    }\n    .scaling-squares-spinner .square:nth-child(4) {\n      -webkit-animation-name: scaling-squares-spinner-animation-child-4;\n              animation-name: scaling-squares-spinner-animation-child-4;\n    }\n    @-webkit-keyframes scaling-squares-animation {\n      50% {\n        -webkit-transform: rotate(90deg);\n                transform: rotate(90deg);\n      }\n      100% {\n        -webkit-transform: rotate(180deg);\n                transform: rotate(180deg);\n      }\n    }\n    @keyframes scaling-squares-animation {\n      50% {\n        -webkit-transform: rotate(90deg);\n                transform: rotate(90deg);\n      }\n      100% {\n        -webkit-transform: rotate(180deg);\n                transform: rotate(180deg);\n      }\n    }\n    @-webkit-keyframes scaling-squares-spinner-animation-child-1 {\n      50% {\n        -webkit-transform: translate(150%,150%) scale(2,2);\n                transform: translate(150%,150%) scale(2,2);\n      }\n    }\n    @keyframes scaling-squares-spinner-animation-child-1 {\n      50% {\n        -webkit-transform: translate(150%,150%) scale(2,2);\n                transform: translate(150%,150%) scale(2,2);\n      }\n    }\n    @-webkit-keyframes scaling-squares-spinner-animation-child-2 {\n      50% {\n        -webkit-transform: translate(-150%,150%) scale(2,2);\n                transform: translate(-150%,150%) scale(2,2);\n      }\n    }\n    @keyframes scaling-squares-spinner-animation-child-2 {\n      50% {\n        -webkit-transform: translate(-150%,150%) scale(2,2);\n                transform: translate(-150%,150%) scale(2,2);\n      }\n    }\n    @-webkit-keyframes scaling-squares-spinner-animation-child-3 {\n      50% {\n        -webkit-transform: translate(-150%,-150%) scale(2,2);\n                transform: translate(-150%,-150%) scale(2,2);\n      }\n    }\n    @keyframes scaling-squares-spinner-animation-child-3 {\n      50% {\n        -webkit-transform: translate(-150%,-150%) scale(2,2);\n                transform: translate(-150%,-150%) scale(2,2);\n      }\n    }\n    @-webkit-keyframes scaling-squares-spinner-animation-child-4 {\n      50% {\n        -webkit-transform: translate(150%,-150%) scale(2,2);\n                transform: translate(150%,-150%) scale(2,2);\n      }\n    }\n    @keyframes scaling-squares-spinner-animation-child-4 {\n      50% {\n        -webkit-transform: translate(150%,-150%) scale(2,2);\n                transform: translate(150%,-150%) scale(2,2);\n      }\n    }\n  "]
            },] },
];
/**
 * @nocollapse
 */
ScalingSquaresSpinnerComponent.ctorParameters = function () { return []; };
ScalingSquaresSpinnerComponent.propDecorators = {
    'size': [{ type: Input },],
    'color': [{ type: Input },],
    'animationDuration': [{ type: Input },],
};
var ScalingSquaresSpinnerModule = /** @class */ (function () {
    function ScalingSquaresSpinnerModule() {
    }
    return ScalingSquaresSpinnerModule;
}());
ScalingSquaresSpinnerModule.decorators = [
    { type: NgModule, args: [{
                imports: [CommonModule],
                exports: [ScalingSquaresSpinnerComponent],
                declarations: [ScalingSquaresSpinnerComponent],
                providers: [],
            },] },
];
/**
 * @nocollapse
 */
ScalingSquaresSpinnerModule.ctorParameters = function () { return []; };
var SelfBuildingSquareSpinnerComponent = /** @class */ (function () {
    function SelfBuildingSquareSpinnerComponent() {
        this.animationDuration = 6000;
        this.size = 40;
        this.color = '#fff';
        this.squaresNum = 9;
    }
    Object.defineProperty(SelfBuildingSquareSpinnerComponent.prototype, "squareSize", {
        /**
         * @return {?}
         */
        get: function () {
            return this.size / 4;
        },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(SelfBuildingSquareSpinnerComponent.prototype, "initialTopPosition", {
        /**
         * @return {?}
         */
        get: function () {
            return -this.squareSize * 2 / 3;
        },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(SelfBuildingSquareSpinnerComponent.prototype, "spinnerStyle", {
        /**
         * @return {?}
         */
        get: function () {
            return {
                top: -this.initialTopPosition + "px",
                height: this.size + "px",
                width: this.size + "px"
            };
        },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(SelfBuildingSquareSpinnerComponent.prototype, "squareStyle", {
        /**
         * @return {?}
         */
        get: function () {
            return {
                height: this.squareSize + "px",
                width: this.squareSize + "px",
                top: this.initialTopPosition + "px",
                marginRight: this.squareSize / 3 + "px",
                marginTop: this.squareSize / 3 + "px",
                animationDuration: this.animationDuration + "ms",
                background: this.color
            };
        },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(SelfBuildingSquareSpinnerComponent.prototype, "squaresStyles", {
        /**
         * @return {?}
         */
        get: function () {
            var /** @type {?} */ squaresStyles = [];
            var /** @type {?} */ delaysMultipliers = [6, 7, 8, 3, 4, 5, 0, 1, 2];
            var /** @type {?} */ delayModifier = this.animationDuration * 0.05;
            for (var /** @type {?} */ i = 0; i < this.squaresNum; i++) {
                squaresStyles.push(Object.assign({ animationDelay: delayModifier * delaysMultipliers[i] + "ms" }, this.squareStyle));
            }
            return squaresStyles;
        },
        enumerable: true,
        configurable: true
    });
    /**
     * @return {?}
     */
    SelfBuildingSquareSpinnerComponent.prototype.ngOnInit = function () {
    };
    return SelfBuildingSquareSpinnerComponent;
}());
SelfBuildingSquareSpinnerComponent.decorators = [
    { type: Component, args: [{
                selector: 'app-self-building-square-spinner',
                template: "\n    <div class=\"self-building-square-spinner\" [ngStyle]=\"spinnerStyle\">\n      <div class=\"square\"\n           *ngFor=\"let ss of squaresStyles; let i = index;\"\n           [ngStyle]=\"ss\"\n           [ngClass]=\"{'clear': i && i % 3 === 0}\"\n      >\n      </div>\n    </div>\n  ",
                styles: ["\n    .self-building-square-spinner, .self-building-square-spinner * {\n      -webkit-box-sizing: border-box;\n              box-sizing: border-box;\n    }\n    .self-building-square-spinner {\n      height: 40px;\n      width: 40px;\n      top: calc( -10px * 2 / 3);\n    }\n    .self-building-square-spinner .square {\n      height: 10px;\n      width: 10px;\n      top: calc( -10px * 2 / 3);\n      margin-right: calc(10px / 3);\n      margin-top: calc(10px / 3);\n      background: #ff1d5e;\n      float: left;\n      position:relative;\n      opacity: 0;\n      -webkit-animation: self-building-square-spinner 6s infinite;\n              animation: self-building-square-spinner 6s infinite;\n    }\n    .self-building-square-spinner .square:nth-child(1) {\n      -webkit-animation-delay: calc(300ms * 6);\n              animation-delay: calc(300ms * 6);\n    }\n    .self-building-square-spinner .square:nth-child(2) {\n      -webkit-animation-delay: calc(300ms * 7);\n              animation-delay: calc(300ms * 7);\n    }\n    .self-building-square-spinner .square:nth-child(3) {\n      -webkit-animation-delay: calc(300ms * 8);\n              animation-delay: calc(300ms * 8);\n    }\n    .self-building-square-spinner .square:nth-child(4) {\n      -webkit-animation-delay: calc(300ms * 3);\n              animation-delay: calc(300ms * 3);\n    }\n    .self-building-square-spinner .square:nth-child(5) {\n      -webkit-animation-delay: calc(300ms * 4);\n              animation-delay: calc(300ms * 4);\n    }\n    .self-building-square-spinner .square:nth-child(6) {\n      -webkit-animation-delay: calc(300ms * 5);\n              animation-delay: calc(300ms * 5);\n    }\n    .self-building-square-spinner .square:nth-child(7) {\n      -webkit-animation-delay: calc(300ms * 0);\n              animation-delay: calc(300ms * 0);\n    }\n    .self-building-square-spinner .square:nth-child(8) {\n      -webkit-animation-delay: calc(300ms * 1);\n              animation-delay: calc(300ms * 1);\n    }\n    .self-building-square-spinner .square:nth-child(9) {\n      -webkit-animation-delay: calc(300ms * 2);\n              animation-delay: calc(300ms * 2);\n    }\n    .self-building-square-spinner .clear{\n      clear: both;\n    }\n    @-webkit-keyframes self-building-square-spinner {\n      0% {\n        opacity: 0;\n      }\n      5% {\n        opacity: 1;\n        top: 0;\n      }\n      50.9% {\n        opacity: 1;\n        top: 0;\n      }\n      55.9% {\n        opacity: 0;\n        top: inherit;\n      }\n    }\n    @keyframes self-building-square-spinner {\n      0% {\n        opacity: 0;\n      }\n      5% {\n        opacity: 1;\n        top: 0;\n      }\n      50.9% {\n        opacity: 1;\n        top: 0;\n      }\n      55.9% {\n        opacity: 0;\n        top: inherit;\n      }\n    }\n  "]
            },] },
];
/**
 * @nocollapse
 */
SelfBuildingSquareSpinnerComponent.ctorParameters = function () { return []; };
SelfBuildingSquareSpinnerComponent.propDecorators = {
    'animationDuration': [{ type: Input },],
    'size': [{ type: Input },],
    'color': [{ type: Input },],
};
var SelfBuildingSquareSpinnerModule = /** @class */ (function () {
    function SelfBuildingSquareSpinnerModule() {
    }
    return SelfBuildingSquareSpinnerModule;
}());
SelfBuildingSquareSpinnerModule.decorators = [
    { type: NgModule, args: [{
                imports: [CommonModule],
                exports: [SelfBuildingSquareSpinnerComponent],
                declarations: [SelfBuildingSquareSpinnerComponent],
                providers: [],
            },] },
];
/**
 * @nocollapse
 */
SelfBuildingSquareSpinnerModule.ctorParameters = function () { return []; };
var SemipolarSpinnerComponent = /** @class */ (function () {
    function SemipolarSpinnerComponent() {
        this.animationDuration = 2000;
        this.size = 65;
        this.color = '#fff';
        this.ringsNum = 5;
    }
    Object.defineProperty(SemipolarSpinnerComponent.prototype, "spinnerStyle", {
        /**
         * @return {?}
         */
        get: function () {
            return {
                height: this.size + "px",
                width: this.size + "px"
            };
        },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(SemipolarSpinnerComponent.prototype, "ringStyle", {
        /**
         * @return {?}
         */
        get: function () {
            return {
                animationDuration: this.animationDuration + "ms",
                borderTopColor: this.color,
                borderLeftColor: this.color
            };
        },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(SemipolarSpinnerComponent.prototype, "ringsStyles", {
        /**
         * @return {?}
         */
        get: function () {
            var /** @type {?} */ ringsStyles = [];
            var /** @type {?} */ delayModifier = 0.1;
            var /** @type {?} */ ringWidth = this.size * 0.05;
            var /** @type {?} */ positionIncrement = ringWidth * 2;
            var /** @type {?} */ sizeDecrement = this.size * 0.2;
            for (var /** @type {?} */ i = 0; i < this.ringsNum; i++) {
                var /** @type {?} */ computedSize = this.size - sizeDecrement * i + "px";
                var /** @type {?} */ computedPosition = positionIncrement * i + "px";
                var /** @type {?} */ style = Object.assign({ animationDelay: this.animationDuration * delayModifier * (this.ringsNum - i - 1) + "ms", height: computedSize, width: computedSize, left: computedPosition, top: computedPosition, borderWidth: ringWidth + "px" }, this.ringStyle);
                ringsStyles.push(style);
            }
            return ringsStyles;
        },
        enumerable: true,
        configurable: true
    });
    /**
     * @return {?}
     */
    SemipolarSpinnerComponent.prototype.ngOnInit = function () {
    };
    return SemipolarSpinnerComponent;
}());
SemipolarSpinnerComponent.decorators = [
    { type: Component, args: [{
                selector: 'app-semipolar-spinner',
                template: "\n    <div class=\"semipolar-spinner\" [ngStyle]=\"spinnerStyle\">\n      <div class=\"ring\" *ngFor=\"let rs of ringsStyles\" [ngStyle]=\"rs\"></div>\n    </div>\n  ",
                styles: ["\n    .semipolar-spinner, .semipolar-spinner * {\n      -webkit-box-sizing: border-box;\n              box-sizing: border-box;\n    }\n    .semipolar-spinner {\n      height: 65px;\n      width: 65px;\n      position: relative;\n    }\n    .semipolar-spinner .ring {\n      border-radius: 50%;\n      position: absolute;\n      border: calc(65px * 0.05) solid transparent;\n      border-top-color: #ff1d5e;\n      border-left-color: #ff1d5e;\n      -webkit-animation: semipolar-spinner-animation 2s infinite;\n              animation: semipolar-spinner-animation 2s infinite;\n    }\n    .semipolar-spinner .ring:nth-child(1) {\n      height: calc(65px - 65px * 0.2 * 0);\n      width: calc(65px - 65px * 0.2 * 0);\n      top: calc(65px * 0.1 * 0);\n      left: calc(65px * 0.1 * 0);\n      -webkit-animation-delay: calc(2000ms * 0.1 * 4);\n              animation-delay: calc(2000ms * 0.1 * 4);\n      z-index: 5;\n    }\n    .semipolar-spinner .ring:nth-child(2) {\n      height: calc(65px - 65px * 0.2 * 1);\n      width: calc(65px - 65px * 0.2 * 1);\n      top: calc(65px * 0.1 * 1);\n      left: calc(65px * 0.1 * 1);\n      -webkit-animation-delay: calc(2000ms * 0.1 * 3);\n              animation-delay: calc(2000ms * 0.1 * 3);\n      z-index: 4;\n    }\n    .semipolar-spinner .ring:nth-child(3) {\n      height: calc(65px - 65px * 0.2 * 2);\n      width: calc(65px - 65px * 0.2 * 2);\n      top: calc(65px * 0.1 * 2);\n      left: calc(65px * 0.1 * 2);\n      -webkit-animation-delay: calc(2000ms * 0.1 * 2);\n              animation-delay: calc(2000ms * 0.1 * 2);\n      z-index: 3;\n    }\n    .semipolar-spinner .ring:nth-child(4) {\n      height: calc(65px - 65px * 0.2 * 3);\n      width: calc(65px - 65px * 0.2 * 3);\n      top: calc(65px * 0.1 * 3);\n      left: calc(65px * 0.1 * 3);\n      -webkit-animation-delay: calc(2000ms * 0.1 * 1);\n              animation-delay: calc(2000ms * 0.1 * 1);\n      z-index: 2;\n    }\n    .semipolar-spinner .ring:nth-child(5) {\n      height: calc(65px - 65px * 0.2 * 4);\n      width: calc(65px - 65px * 0.2 * 4);\n      top: calc(65px * 0.1 * 4);\n      left: calc(65px * 0.1 * 4);\n      -webkit-animation-delay: calc(2000ms * 0.1 * 0);\n              animation-delay: calc(2000ms * 0.1 * 0);\n      z-index: 1;\n    }\n    @-webkit-keyframes semipolar-spinner-animation {\n      50% {\n        -webkit-transform: rotate(360deg) scale(0.7);\n                transform: rotate(360deg) scale(0.7);\n      }\n    }\n    @keyframes semipolar-spinner-animation {\n      50% {\n        -webkit-transform: rotate(360deg) scale(0.7);\n                transform: rotate(360deg) scale(0.7);\n      }\n    }\n  "]
            },] },
];
/**
 * @nocollapse
 */
SemipolarSpinnerComponent.ctorParameters = function () { return []; };
SemipolarSpinnerComponent.propDecorators = {
    'animationDuration': [{ type: Input },],
    'size': [{ type: Input },],
    'color': [{ type: Input },],
};
var SemipolarSpinnerModule = /** @class */ (function () {
    function SemipolarSpinnerModule() {
    }
    return SemipolarSpinnerModule;
}());
SemipolarSpinnerModule.decorators = [
    { type: NgModule, args: [{
                imports: [CommonModule],
                exports: [SemipolarSpinnerComponent],
                declarations: [SemipolarSpinnerComponent],
                providers: [],
            },] },
];
/**
 * @nocollapse
 */
SemipolarSpinnerModule.ctorParameters = function () { return []; };
var SpringSpinnerComponent = /** @class */ (function () {
    function SpringSpinnerComponent() {
        this.size = 60;
        this.color = '#fff';
        this.animationDuration = 2000;
        this.animationBaseName = 'spring-spinner-animation';
        this.currentAnimationName = '';
    }
    Object.defineProperty(SpringSpinnerComponent.prototype, "spinnerStyle", {
        /**
         * @return {?}
         */
        get: function () {
            return {
                height: this.size + "px",
                width: this.size + "px"
            };
        },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(SpringSpinnerComponent.prototype, "spinnerPartStyle", {
        /**
         * @return {?}
         */
        get: function () {
            return {
                height: this.size / 2 + "px",
                width: this.size + "px"
            };
        },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(SpringSpinnerComponent.prototype, "rotatorStyle", {
        /**
         * @return {?}
         */
        get: function () {
            return {
                height: this.size + "px",
                width: this.size + "px",
                borderRightColor: this.color,
                borderTopColor: this.color,
                borderWidth: this.size / 7 + "px",
                animationDuration: this.animationDuration + "ms",
                animationName: this.currentAnimationName
            };
        },
        enumerable: true,
        configurable: true
    });
    /**
     * @return {?}
     */
    SpringSpinnerComponent.prototype.ngOnInit = function () {
    };
    return SpringSpinnerComponent;
}());
SpringSpinnerComponent.decorators = [
    { type: Component, args: [{
                selector: 'app-spring-spinner',
                template: "\n    <div class=\"spring-spinner\" [ngStyle]=\"spinnerStyle\">\n      <div class=\"spring-spinner-part top\" [ngStyle]=\"spinnerPartStyle\">\n        <div class=\"spring-spinner-rotator\" [ngStyle]=\"rotatorStyle\"></div>\n      </div>\n      <div class=\"spring-spinner-part bottom\" [ngStyle]=\"spinnerPartStyle\">\n        <div class=\"spring-spinner-rotator\" [ngStyle]=\"rotatorStyle\"></div>\n      </div>\n    </div>\n  ",
                styles: ["\n    .spring-spinner, .spring-spinner * {\n      -webkit-box-sizing: border-box;\n              box-sizing: border-box;\n    }\n    .spring-spinner {\n      height: 60px;\n      width: 60px;\n    }\n    .spring-spinner .spring-spinner-part {\n      overflow: hidden;\n      height: calc(60px / 2);\n      width: 60px;\n    }\n    .spring-spinner  .spring-spinner-part.bottom {\n      -webkit-transform: rotate(180deg) scale(-1, 1);\n              transform: rotate(180deg) scale(-1, 1);\n    }\n    .spring-spinner .spring-spinner-rotator {\n      width: 60px;\n      height: 60px;\n      border: calc(60px / 7) solid transparent;\n      border-right-color: #ff1d5e;\n      border-top-color: #ff1d5e;\n      border-radius: 50%;\n      -webkit-box-sizing: border-box;\n              box-sizing: border-box;\n      -webkit-animation: spring-spinner-animation 3s ease-in-out infinite;\n              animation: spring-spinner-animation 3s ease-in-out infinite;\n      -webkit-transform: rotate(-200deg);\n              transform: rotate(-200deg);\n    }\n    @-webkit-keyframes spring-spinner-animation {\n      0% {\n        border-width: calc(60px / 7);\n      }\n      25% {\n        border-width: calc(60px / 23.33);\n      }\n      50% {\n        -webkit-transform: rotate(115deg);\n                transform: rotate(115deg);\n        border-width: calc(60px / 7);\n      }\n      75% {\n        border-width: calc(60px / 23.33);\n      }\n      100% {\n        border-width: calc(60px / 7);\n      }\n    }\n    @keyframes spring-spinner-animation {\n      0% {\n        border-width: calc(60px / 7);\n      }\n      25% {\n        border-width: calc(60px / 23.33);\n      }\n      50% {\n        -webkit-transform: rotate(115deg);\n                transform: rotate(115deg);\n        border-width: calc(60px / 7);\n      }\n      75% {\n        border-width: calc(60px / 23.33);\n      }\n      100% {\n        border-width: calc(60px / 7);\n      }\n    }\n  "]
            },] },
];
/**
 * @nocollapse
 */
SpringSpinnerComponent.ctorParameters = function () { return []; };
SpringSpinnerComponent.propDecorators = {
    'size': [{ type: Input },],
    'color': [{ type: Input },],
    'animationDuration': [{ type: Input },],
};
var SpringSpinnerModule = /** @class */ (function () {
    function SpringSpinnerModule() {
    }
    return SpringSpinnerModule;
}());
SpringSpinnerModule.decorators = [
    { type: NgModule, args: [{
                imports: [CommonModule],
                exports: [SpringSpinnerComponent],
                declarations: [SpringSpinnerComponent],
                providers: [],
            },] },
];
/**
 * @nocollapse
 */
SpringSpinnerModule.ctorParameters = function () { return []; };
var SwappingSquaresSpinnerComponent = /** @class */ (function () {
    function SwappingSquaresSpinnerComponent() {
        this.animationDuration = 1000;
        this.size = 65;
        this.color = '#fff';
        this.animationBaseName = 'swapping-square-animation-child';
        this.squaresNum = 4;
    }
    Object.defineProperty(SwappingSquaresSpinnerComponent.prototype, "spinnerStyle", {
        /**
         * @return {?}
         */
        get: function () {
            return {
                height: this.size + "px",
                width: this.size + "px"
            };
        },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(SwappingSquaresSpinnerComponent.prototype, "squareStyle", {
        /**
         * @return {?}
         */
        get: function () {
            return {
                height: this.size * 0.25 / 1.3 + "px",
                width: this.size * 0.25 / 1.3 + "px",
                animationDuration: this.animationDuration + "ms",
                borderWidth: this.size * 0.04 / 1.3 + "px",
                borderColor: this.color
            };
        },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(SwappingSquaresSpinnerComponent.prototype, "squaresStyles", {
        /**
         * @return {?}
         */
        get: function () {
            var /** @type {?} */ squaresStyles = [];
            var /** @type {?} */ delay = this.animationDuration * 0.5;
            for (var /** @type {?} */ i = 1; i <= this.squaresNum; i++) {
                squaresStyles.push(Object.assign({ animationDelay: (i % 2 === 0 ? delay : 0) + "ms" }, this.squareStyle));
            }
            return squaresStyles;
        },
        enumerable: true,
        configurable: true
    });
    /**
     * @return {?}
     */
    SwappingSquaresSpinnerComponent.prototype.ngOnInit = function () {
    };
    return SwappingSquaresSpinnerComponent;
}());
SwappingSquaresSpinnerComponent.decorators = [
    { type: Component, args: [{
                selector: 'app-swapping-squares-spinner',
                template: "\n    <div class=\"swapping-squares-spinner\" [ngStyle]=\"spinnerStyle\">\n      <div class=\"square\"\n           *ngFor=\"let ss of squaresStyles; let i = index;\"\n           [ngClass]=\"'square-' + (i + 1)\"\n           [ngStyle]=\"ss\">\n      </div>\n    </div>\n  ",
                styles: ["\n    .swapping-squares-spinner, .swapping-squares-spinner * {\n      -webkit-box-sizing: border-box;\n              box-sizing: border-box;\n    }\n    .swapping-squares-spinner {\n      height: 65px;\n      width: 65px;\n      position: relative;\n      display: -webkit-box;\n      display: -ms-flexbox;\n      display: flex;\n      -webkit-box-orient: horizontal;\n      -webkit-box-direction: normal;\n          -ms-flex-direction: row;\n              flex-direction: row;\n      -webkit-box-pack: center;\n          -ms-flex-pack: center;\n              justify-content: center;\n      -webkit-box-align: center;\n          -ms-flex-align: center;\n              align-items: center;\n    }\n    .swapping-squares-spinner .square {\n      height: calc(65px * 0.25 / 1.3);\n      width:  calc(65px * 0.25 / 1.3);\n      -webkit-animation-duration: 1000ms;\n              animation-duration: 1000ms;\n      border: calc(65px * 0.04 / 1.3) solid #ff1d5e;\n      margin-right: auto;\n      margin-left: auto;\n      position: absolute;\n      -webkit-animation-iteration-count: infinite;\n              animation-iteration-count: infinite;\n    }\n    .swapping-squares-spinner .square:nth-child(1) {\n      -webkit-animation-name: swapping-squares-animation-child-1;\n              animation-name: swapping-squares-animation-child-1;\n      -webkit-animation-delay: 500ms;\n              animation-delay: 500ms;\n    }\n    .swapping-squares-spinner .square:nth-child(2) {\n      -webkit-animation-name: swapping-squares-animation-child-2;\n              animation-name: swapping-squares-animation-child-2;\n      -webkit-animation-delay: 0ms;\n              animation-delay: 0ms;\n    }\n    .swapping-squares-spinner .square:nth-child(3) {\n      -webkit-animation-name: swapping-squares-animation-child-3;\n              animation-name: swapping-squares-animation-child-3;\n      -webkit-animation-delay: 500ms;\n              animation-delay: 500ms;\n    }\n    .swapping-squares-spinner .square:nth-child(4) {\n      -webkit-animation-name: swapping-squares-animation-child-4;\n              animation-name: swapping-squares-animation-child-4;\n      -webkit-animation-delay: 0ms;\n              animation-delay: 0ms;\n    }\n    @-webkit-keyframes swapping-squares-animation-child-1 {\n      50% {\n        -webkit-transform: translate(150%,150%) scale(2,2);\n                transform: translate(150%,150%) scale(2,2);\n      }\n    }\n    @keyframes swapping-squares-animation-child-1 {\n      50% {\n        -webkit-transform: translate(150%,150%) scale(2,2);\n                transform: translate(150%,150%) scale(2,2);\n      }\n    }\n    @-webkit-keyframes swapping-squares-animation-child-2 {\n      50% {\n        -webkit-transform: translate(-150%,150%) scale(2,2);\n                transform: translate(-150%,150%) scale(2,2);\n      }\n    }\n    @keyframes swapping-squares-animation-child-2 {\n      50% {\n        -webkit-transform: translate(-150%,150%) scale(2,2);\n                transform: translate(-150%,150%) scale(2,2);\n      }\n    }\n    @-webkit-keyframes swapping-squares-animation-child-3 {\n      50% {\n        -webkit-transform: translate(-150%,-150%) scale(2,2);\n                transform: translate(-150%,-150%) scale(2,2);\n      }\n    }\n    @keyframes swapping-squares-animation-child-3 {\n      50% {\n        -webkit-transform: translate(-150%,-150%) scale(2,2);\n                transform: translate(-150%,-150%) scale(2,2);\n      }\n    }\n    @-webkit-keyframes swapping-squares-animation-child-4 {\n      50% {\n        -webkit-transform: translate(150%,-150%) scale(2,2);\n                transform: translate(150%,-150%) scale(2,2);\n      }\n    }\n    @keyframes swapping-squares-animation-child-4 {\n      50% {\n        -webkit-transform: translate(150%,-150%) scale(2,2);\n                transform: translate(150%,-150%) scale(2,2);\n      }\n    }\n  "]
            },] },
];
/**
 * @nocollapse
 */
SwappingSquaresSpinnerComponent.ctorParameters = function () { return []; };
SwappingSquaresSpinnerComponent.propDecorators = {
    'animationDuration': [{ type: Input },],
    'size': [{ type: Input },],
    'color': [{ type: Input },],
};
var SwappingSquaresSpinnerModule = /** @class */ (function () {
    function SwappingSquaresSpinnerModule() {
    }
    return SwappingSquaresSpinnerModule;
}());
SwappingSquaresSpinnerModule.decorators = [
    { type: NgModule, args: [{
                imports: [CommonModule],
                exports: [SwappingSquaresSpinnerComponent],
                declarations: [SwappingSquaresSpinnerComponent],
                providers: [],
            },] },
];
/**
 * @nocollapse
 */
SwappingSquaresSpinnerModule.ctorParameters = function () { return []; };
var TrinityRingsSpinnerComponent = /** @class */ (function () {
    function TrinityRingsSpinnerComponent() {
        this.containerPadding = 3;
        this.animationDuration = 1500;
        this.size = 60;
        this.color = '#fff';
    }
    Object.defineProperty(TrinityRingsSpinnerComponent.prototype, "outerRingSize", {
        /**
         * @return {?}
         */
        get: function () {
            return this.size - this.containerPadding * 2;
        },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(TrinityRingsSpinnerComponent.prototype, "spinnerStyle", {
        /**
         * @return {?}
         */
        get: function () {
            return {
                height: this.size + "px",
                width: this.size + "px",
                padding: this.containerPadding + "px"
            };
        },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(TrinityRingsSpinnerComponent.prototype, "ring1Style", {
        /**
         * @return {?}
         */
        get: function () {
            return {
                height: this.outerRingSize + "px",
                width: this.outerRingSize + "px",
                borderColor: this.color,
                animationDuration: this.animationDuration + "ms"
            };
        },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(TrinityRingsSpinnerComponent.prototype, "ring2Style", {
        /**
         * @return {?}
         */
        get: function () {
            return {
                height: this.outerRingSize * 0.65 + "px",
                width: this.outerRingSize * 0.65 + "px",
                borderColor: this.color,
                animationDuration: this.animationDuration + "ms"
            };
        },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(TrinityRingsSpinnerComponent.prototype, "ring3Style", {
        /**
         * @return {?}
         */
        get: function () {
            return {
                height: this.outerRingSize * 0.1 + "px",
                width: this.outerRingSize * 0.1 + "px",
                borderColor: this.color,
                animationDuration: this.animationDuration + "ms"
            };
        },
        enumerable: true,
        configurable: true
    });
    /**
     * @return {?}
     */
    TrinityRingsSpinnerComponent.prototype.ngOnInit = function () {
    };
    return TrinityRingsSpinnerComponent;
}());
TrinityRingsSpinnerComponent.decorators = [
    { type: Component, args: [{
                selector: 'app-trinity-rings-spinner',
                template: "\n    <div class=\"trinity-rings-spinner\" [ngStyle]=\"spinnerStyle\">\n      <div class=\"circle circle1\" [ngStyle]=\"ring1Style\"></div>\n      <div class=\"circle circle3\" [ngStyle]=\"ring3Style\"></div>\n      <div class=\"circle circle2\" [ngStyle]=\"ring2Style\"></div>\n    </div>\n  ",
                styles: ["\n    .trinity-rings-spinner, .trinity-rings-spinner * {\n      -webkit-box-sizing: border-box;\n              box-sizing: border-box;\n    }\n    .trinity-rings-spinner {\n      height: 66px;\n      width: 66px;\n      padding: 3px;\n      position: relative;\n      display: -webkit-box;\n      display: -ms-flexbox;\n      display: flex;\n      -webkit-box-pack: center;\n          -ms-flex-pack: center;\n              justify-content: center;\n      -webkit-box-align: center;\n          -ms-flex-align: center;\n              align-items: center;\n      -webkit-box-orient: horizontal;\n      -webkit-box-direction: normal;\n          -ms-flex-direction: row;\n              flex-direction: row;\n      overflow: hidden;\n      -webkit-box-sizing: border-box;\n              box-sizing: border-box;\n    }\n    .trinity-rings-spinner .circle {\n      position:absolute;\n      display:block;\n      border-radius:50%;\n      border: 3px solid #ff1d5e;\n      opacity: 1;\n    }\n    .trinity-rings-spinner .circle:nth-child(1) {\n      height: 60px;\n      width: 60px;\n      -webkit-animation : trinity-rings-spinner-circle1-animation 1.5s infinite linear;\n              animation : trinity-rings-spinner-circle1-animation 1.5s infinite linear;\n      border-width: 3px;\n    }\n    .trinity-rings-spinner .circle:nth-child(2) {\n      height: calc(60px * 0.65);\n      width: calc(60px * 0.65);\n      -webkit-animation : trinity-rings-spinner-circle2-animation 1.5s infinite linear;\n              animation : trinity-rings-spinner-circle2-animation 1.5s infinite linear;\n      border-width: 2px;\n    }\n    .trinity-rings-spinner .circle:nth-child(3) {\n      height: calc(60px * 0.1);\n      width: calc(60px * 0.1);\n      -webkit-animation:trinity-rings-spinner-circle3-animation 1.5s infinite linear;\n              animation:trinity-rings-spinner-circle3-animation 1.5s infinite linear;\n      border-width: 1px;\n    }\n    @-webkit-keyframes trinity-rings-spinner-circle1-animation{\n      0% {\n        -webkit-transform: rotateZ(20deg) rotateY(0deg);\n                transform: rotateZ(20deg) rotateY(0deg);\n      }\n      100% {\n        -webkit-transform: rotateZ(100deg) rotateY(360deg);\n                transform: rotateZ(100deg) rotateY(360deg);\n      }\n    }\n    @keyframes trinity-rings-spinner-circle1-animation{\n      0% {\n        -webkit-transform: rotateZ(20deg) rotateY(0deg);\n                transform: rotateZ(20deg) rotateY(0deg);\n      }\n      100% {\n        -webkit-transform: rotateZ(100deg) rotateY(360deg);\n                transform: rotateZ(100deg) rotateY(360deg);\n      }\n    }\n    @-webkit-keyframes trinity-rings-spinner-circle2-animation{\n      0% {\n        -webkit-transform: rotateZ(100deg) rotateX(0deg);\n                transform: rotateZ(100deg) rotateX(0deg);\n      }\n      100% {\n        -webkit-transform: rotateZ(0deg) rotateX(360deg);\n                transform: rotateZ(0deg) rotateX(360deg);\n      }\n    }\n    @keyframes trinity-rings-spinner-circle2-animation{\n      0% {\n        -webkit-transform: rotateZ(100deg) rotateX(0deg);\n                transform: rotateZ(100deg) rotateX(0deg);\n      }\n      100% {\n        -webkit-transform: rotateZ(0deg) rotateX(360deg);\n                transform: rotateZ(0deg) rotateX(360deg);\n      }\n    }\n    @-webkit-keyframes trinity-rings-spinner-circle3-animation{\n      0% {\n        -webkit-transform: rotateZ(100deg) rotateX(-360deg);\n                transform: rotateZ(100deg) rotateX(-360deg);\n      }\n      100% {\n        -webkit-transform: rotateZ(-360deg) rotateX(360deg);\n                transform: rotateZ(-360deg) rotateX(360deg);\n      }\n    }\n    @keyframes trinity-rings-spinner-circle3-animation{\n      0% {\n        -webkit-transform: rotateZ(100deg) rotateX(-360deg);\n                transform: rotateZ(100deg) rotateX(-360deg);\n      }\n      100% {\n        -webkit-transform: rotateZ(-360deg) rotateX(360deg);\n                transform: rotateZ(-360deg) rotateX(360deg);\n      }\n    }\n  "]
            },] },
];
/**
 * @nocollapse
 */
TrinityRingsSpinnerComponent.ctorParameters = function () { return []; };
TrinityRingsSpinnerComponent.propDecorators = {
    'animationDuration': [{ type: Input },],
    'size': [{ type: Input },],
    'color': [{ type: Input },],
};
var TrinityRingsSpinnerModule = /** @class */ (function () {
    function TrinityRingsSpinnerModule() {
    }
    return TrinityRingsSpinnerModule;
}());
TrinityRingsSpinnerModule.decorators = [
    { type: NgModule, args: [{
                imports: [CommonModule],
                exports: [TrinityRingsSpinnerComponent],
                declarations: [TrinityRingsSpinnerComponent],
                providers: [],
            },] },
];
/**
 * @nocollapse
 */
TrinityRingsSpinnerModule.ctorParameters = function () { return []; };
/**
 * Generated bundle index. Do not edit.
 */
export { AtomSpinnerModule, BreedingRhombusSpinnerModule, CirclesToRhumbusesSpinnerModule, FingerprintSpinnerModule, FulfillingBouncingCircleSpinnerModule, FlowerSpinnerModule, FulfillingSquareSpinnerModule, HalfCircleSpinnerModule, HollowDotsSpinnerModule, IntersectingCirclesSpinnerModule, LoopingRhumbusesSpinnerModule, OrbitSpinnerModule, PixelSpinnerModule, RadarSpinnerModule, ScalingSquaresSpinnerModule, SelfBuildingSquareSpinnerModule, SemipolarSpinnerModule, SpringSpinnerModule, SwappingSquaresSpinnerModule, TrinityRingsSpinnerModule, AtomSpinnerComponent as a, BreedingRhombusSpinnerComponent as b, CirclesToRhumbusesSpinnerComponent as c, FingerprintSpinnerComponent as d, FlowerSpinnerComponent as f, FulfillingBouncingCircleSpinnerComponent as e, FulfillingSquareSpinnerComponent as g, HalfCircleSpinnerComponent as h, HollowDotsSpinnerComponent as i, IntersectingCirclesSpinnerComponent as j, LoopingRhumbusesSpinnerComponent as k, OrbitSpinnerComponent as l, PixelSpinnerComponent as m, RadarSpinnerComponent as n, ScalingSquaresSpinnerComponent as o, SelfBuildingSquareSpinnerComponent as p, SemipolarSpinnerComponent as q, SpringSpinnerComponent as r, SwappingSquaresSpinnerComponent as s, TrinityRingsSpinnerComponent as t };
